{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from scholarly import scholarly\n",
    "\n",
    "search_query = scholarly.search_author('Jens Meiler')\n",
    "#author = next(search_query, None)\n",
    "#scholarly.fill(author)\n",
    "\n",
    "# not working because google blocked me"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from tqdm import tqdm\n",
    "import requests\n",
    "\n",
    "jens_orcid = '0000-0001-8945-193X'\n",
    "\n",
    "orcid_api_url = f\"https://pub.orcid.org/v3.0/{jens_orcid}/works\"\n",
    "headers = {\"Accept\": \"application/json\"}\n",
    "\n",
    "response = requests.get(orcid_api_url, headers=headers)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "267 of 298\n"
     ]
    }
   ],
   "source": [
    "data = response.json()\n",
    "works = data.get('group', [])\n",
    "\n",
    "auto_dois = []\n",
    "\n",
    "for work in works:\n",
    "        work_summary = work.get('work-summary', [])\n",
    "        for summary in work_summary:\n",
    "            if 'external-ids' in summary:\n",
    "                external_ids = summary['external-ids']['external-id']\n",
    "                for external_id in external_ids:\n",
    "                    if external_id.get('external-id-type') == 'doi':\n",
    "                        auto_dois.append('https://doi.org/' + external_id.get('external-id-value'))\n",
    "\n",
    "print(len(auto_dois), 'of', len(works))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "manual_dois = \"\"\"https://doi.org/10.1101/2024.08.06.606224 \n",
    "\n",
    "https://doi.org/10.48550/arXiv.2411.09820 \n",
    "\n",
    "https://doi.org/10.1101/2024.06.20.599843 \n",
    "\n",
    "https://doi.org/10.26434/chemrxiv-2024-hr692 \n",
    "\n",
    "https://doi.org/10.1021/acschemneuro.4c00086 \n",
    "\n",
    "https://doi.org/10.1021/acssynbio.3c00753 \n",
    "\n",
    "https://doi.org/10.1371/journal.pcbi.1011939 \n",
    "\n",
    "https://doi.org/10.1021/acs.jctc.3c01081 \n",
    "\n",
    "https://doi.org/10.1101/2022.11.15.516482  \n",
    "\n",
    "https://doi.org/10.1016/j.sbi.2023.102645\n",
    "\n",
    "https://doi.org/10.1609/aaai.v37i12.26679 \n",
    "\n",
    "https://doi.org/10.1016/j.bpj.2022.11.011 \n",
    "\n",
    "https://doi.org/10.1101/2023.04.17.537185 \n",
    "\n",
    "https://doi.org/10.3389/fmolb.2023.1121962 \n",
    "\n",
    "https://doi.org/10.1016/j.str.2022.11.012 \n",
    "\n",
    "https://doi.org/10.1073/pnas.2206129119 \n",
    "\n",
    "https://doi.org/10.1158/0008-5472.CAN-21-3798 \n",
    "\n",
    "https://doi.org/10.7554/eLife.75751\n",
    "\n",
    "https://doi.org/10.3389/fphar.2022.833099 \n",
    "\n",
    "https://doi.org/10.1021/acs.jproteome.1c00410 \n",
    "\n",
    "https://doi.org/10.1021/acs.jcim.0c01001 \n",
    "\n",
    "https://doi.org/10.1021/acs.jcim.0c01001 \n",
    "\n",
    "https://doi.org/10.1109/3DV50981.2020.00103\n",
    "\n",
    "https://doi.org/10.48550/arXiv.2007.07029 \n",
    "\n",
    "https://doi.org/10.1007/s10822-019-00199-8 \n",
    "\n",
    "https://doi.org/10.3390/molecules22091576 \n",
    "\n",
    "https://doi.org/10.1016/j.bmcl.2016.07.071\n",
    "\n",
    "https://doi.org/10.1007/s10822-016-9895-2 \n",
    "\n",
    "https://doi.org/10.1007/s10822-015-9893-9 \n",
    "\n",
    "https://doi.org/10.1124/pr.112.007336 \n",
    "\n",
    "https://doi.org/10.3390/molecules18010735 \n",
    "\n",
    "https://doi.org/10.3390/molecules17089971 \n",
    "\n",
    "https://doi.org/10.1109/CIBCB.2012.6217248\n",
    "\n",
    "https://doi.org/10.1109/CIBCB.2012.6217246\n",
    "\n",
    "https://doi.org/10.1109/CIBCB.2011.5948478\n",
    "\n",
    "https://doi.org/10.1002/cmdc.201100510 \n",
    "\n",
    "https://doi.org/10.1109/CIBCB.2011.5948478\n",
    "\n",
    "https://doi.org/10.1016/j.bpj.2010.12.2344 \n",
    "\n",
    "https://doi.org/10.1016/j.bmcl.2010.07.061\n",
    "\n",
    "https://doi.org/10.1109/CIBCB.2009.4925736\n",
    "\n",
    "https://doi.org/10.1109/CIBCB.2009.4925709\n",
    "\n",
    "https://doi.org/10.1002/qsar.200330837 \n",
    "\n",
    "https://doi.org/10.1023/A:1023060720156 \n",
    "\n",
    "https://doi.org/10.1006/jmre.2002.2599\n",
    "\n",
    "https://doi.org/10.1021/ci0102970 \n",
    "\n",
    "https://doi.org/10.1021/ci000021c\n",
    "\n",
    "https://doi.org/10.1007/PL00010287\"\"\"\n",
    "\n",
    "manual_dois = manual_dois.split()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "45"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(manual_dois))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "291 16\n"
     ]
    }
   ],
   "source": [
    "joint_dois = set(manual_dois)\n",
    "joint_dois.update(auto_dois)\n",
    "joint_dois = list(joint_dois)\n",
    "intersect_dois = list(set(manual_dois) & set(auto_dois))\n",
    "print(len(joint_dois), len(intersect_dois))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'indexed': {'date-parts': [[2024, 10, 29]],\n",
       "  'date-time': '2024-10-29T09:11:08Z',\n",
       "  'timestamp': 1730193068708,\n",
       "  'version': '3.28.0'},\n",
       " 'reference-count': 28,\n",
       " 'publisher': 'IEEE',\n",
       " 'license': [{'start': {'date-parts': [[2020, 11, 1]],\n",
       "    'date-time': '2020-11-01T00:00:00Z',\n",
       "    'timestamp': 1604188800000},\n",
       "   'content-version': 'vor',\n",
       "   'delay-in-days': 0,\n",
       "   'URL': 'https://ieeexplore.ieee.org/Xplorehelp/downloads/license-information/IEEE.html'},\n",
       "  {'start': {'date-parts': [[2020, 11, 1]],\n",
       "    'date-time': '2020-11-01T00:00:00Z',\n",
       "    'timestamp': 1604188800000},\n",
       "   'content-version': 'stm-asf',\n",
       "   'delay-in-days': 0,\n",
       "   'URL': 'https://doi.org/10.15223/policy-029'},\n",
       "  {'start': {'date-parts': [[2020, 11, 1]],\n",
       "    'date-time': '2020-11-01T00:00:00Z',\n",
       "    'timestamp': 1604188800000},\n",
       "   'content-version': 'stm-asf',\n",
       "   'delay-in-days': 0,\n",
       "   'URL': 'https://doi.org/10.15223/policy-037'}],\n",
       " 'content-domain': {'domain': [], 'crossmark-restriction': False},\n",
       " 'published-print': {'date-parts': [[2020, 11]]},\n",
       " 'DOI': '10.1109/3dv50981.2020.00103',\n",
       " 'type': 'proceedings-article',\n",
       " 'created': {'date-parts': [[2021, 3, 15]],\n",
       "  'date-time': '2021-03-15T21:00:57Z',\n",
       "  'timestamp': 1615842057000},\n",
       " 'page': '928-937',\n",
       " 'source': 'Crossref',\n",
       " 'is-referenced-by-count': 4,\n",
       " 'title': '3D Deep Learning for Biological Function Prediction from Physical Fields',\n",
       " 'prefix': '10.1109',\n",
       " 'author': [{'given': 'Vladimir',\n",
       "   'family': 'Golkov',\n",
       "   'sequence': 'first',\n",
       "   'affiliation': []},\n",
       "  {'given': 'Marcin J.',\n",
       "   'family': 'Skwark',\n",
       "   'sequence': 'additional',\n",
       "   'affiliation': []},\n",
       "  {'given': 'Atanas',\n",
       "   'family': 'Mirchev',\n",
       "   'sequence': 'additional',\n",
       "   'affiliation': []},\n",
       "  {'given': 'Georgi',\n",
       "   'family': 'Dikov',\n",
       "   'sequence': 'additional',\n",
       "   'affiliation': []},\n",
       "  {'given': 'Alexander R.',\n",
       "   'family': 'Geanes',\n",
       "   'sequence': 'additional',\n",
       "   'affiliation': []},\n",
       "  {'given': 'Jeffrey',\n",
       "   'family': 'Mendenhall',\n",
       "   'sequence': 'additional',\n",
       "   'affiliation': []},\n",
       "  {'given': 'Jens',\n",
       "   'family': 'Meiler',\n",
       "   'sequence': 'additional',\n",
       "   'affiliation': []},\n",
       "  {'given': 'Daniel',\n",
       "   'family': 'Cremers',\n",
       "   'sequence': 'additional',\n",
       "   'affiliation': []}],\n",
       " 'member': '263',\n",
       " 'reference': [{'doi-asserted-by': 'publisher',\n",
       "   'key': 'ref10',\n",
       "   'DOI': '10.1093/bioinformatics/bth365'},\n",
       "  {'doi-asserted-by': 'publisher',\n",
       "   'key': 'ref11',\n",
       "   'DOI': '10.1016/S0040-4039(01)94977-9'},\n",
       "  {'key': 'ref12',\n",
       "   'first-page': '6488',\n",
       "   'article-title': 'Protein contact prediction from amino acid coevolution using convolutional networks for graph-valued images',\n",
       "   'volume': '29',\n",
       "   'author': 'golkov',\n",
       "   'year': '2016',\n",
       "   'journal-title': 'Adv Neural Inf Process Syst (NIPS)'},\n",
       "  {'year': '2011',\n",
       "   'author': 'israelachvili',\n",
       "   'journal-title': 'Intermolecular and Surface Forces',\n",
       "   'key': 'ref13'},\n",
       "  {'doi-asserted-by': 'publisher',\n",
       "   'key': 'ref14',\n",
       "   'DOI': '10.1186/s13059-016-1037-6'},\n",
       "  {'year': '2015',\n",
       "   'author': 'kingma',\n",
       "   'journal-title': 'Int Conf on Learning Representations (ICLR)',\n",
       "   'article-title': 'Adam: a method for stochastic optimization',\n",
       "   'key': 'ref15'},\n",
       "  {'key': 'ref16',\n",
       "   'first-page': '4824',\n",
       "   'article-title': 'ImageNet classification with deep convolutional neural networks',\n",
       "   'volume': '25',\n",
       "   'author': 'krizhevsky',\n",
       "   'year': '2012',\n",
       "   'journal-title': 'Adv Neural Inf Process Syst (NIPS)'},\n",
       "  {'key': 'ref17',\n",
       "   'first-page': '30',\n",
       "   'article-title': 'Rectifier nonlinearities improve neural network acoustic models',\n",
       "   'author': 'maas',\n",
       "   'year': '2013',\n",
       "   'journal-title': 'Proc of the International Conference on Machine Learning (ICML)'},\n",
       "  {'doi-asserted-by': 'publisher',\n",
       "   'key': 'ref18',\n",
       "   'DOI': '10.1109/ICRA.2015.7139679'},\n",
       "  {'doi-asserted-by': 'publisher',\n",
       "   'key': 'ref19',\n",
       "   'DOI': '10.1109/IROS.2015.7353481'},\n",
       "  {'doi-asserted-by': 'publisher',\n",
       "   'key': 'ref28',\n",
       "   'DOI': '10.1371/journal.pcbi.1005324'},\n",
       "  {'doi-asserted-by': 'publisher',\n",
       "   'key': 'ref4',\n",
       "   'DOI': '10.1107/S0021889809054326'},\n",
       "  {'doi-asserted-by': 'publisher',\n",
       "   'key': 'ref27',\n",
       "   'DOI': '10.1371/journal.pone.0021554'},\n",
       "  {'doi-asserted-by': 'publisher',\n",
       "   'key': 'ref3',\n",
       "   'DOI': '10.1002/1521-3838(200211)21:5<507::AID-QSAR507>3.0.CO;2-L'},\n",
       "  {'doi-asserted-by': 'publisher',\n",
       "   'key': 'ref6',\n",
       "   'DOI': '10.3390/molecules18010735'},\n",
       "  {'doi-asserted-by': 'publisher',\n",
       "   'key': 'ref5',\n",
       "   'DOI': '10.1002/jcc.540040211'},\n",
       "  {'year': '2019',\n",
       "   'author': 'libera',\n",
       "   'journal-title': 'arXiv org arXiv 1910 14594',\n",
       "   'article-title': 'Deep Learning for 2D and 3D Rotatable Data: An Overview of Methods',\n",
       "   'key': 'ref8'},\n",
       "  {'doi-asserted-by': 'publisher', 'key': 'ref7', 'DOI': '10.1021/ci500319f'},\n",
       "  {'doi-asserted-by': 'publisher',\n",
       "   'key': 'ref2',\n",
       "   'DOI': '10.3390/molecules15053281'},\n",
       "  {'doi-asserted-by': 'publisher', 'key': 'ref9', 'DOI': '10.1021/jm030331x'},\n",
       "  {'doi-asserted-by': 'publisher', 'key': 'ref1', 'DOI': '10.1063/1.457198'},\n",
       "  {'doi-asserted-by': 'publisher',\n",
       "   'key': 'ref20',\n",
       "   'DOI': '10.1007/s10822-016-9895-2'},\n",
       "  {'doi-asserted-by': 'publisher', 'key': 'ref22', 'DOI': '10.1002/wcms.1121'},\n",
       "  {'doi-asserted-by': 'publisher', 'key': 'ref21', 'DOI': '10.1021/ci900043r'},\n",
       "  {'doi-asserted-by': 'publisher',\n",
       "   'key': 'ref24',\n",
       "   'DOI': '10.1124/pr.112.007336'},\n",
       "  {'year': '2015',\n",
       "   'author': 'simonyan',\n",
       "   'journal-title': 'Int Conf on Learning Representations (ICLR)',\n",
       "   'article-title': 'Very deep convolutional networks for large-scale image recognition',\n",
       "   'key': 'ref23'},\n",
       "  {'year': '2015',\n",
       "   'author': 'wallach',\n",
       "   'journal-title': 'arXiv org arXiv 1510 02855',\n",
       "   'article-title': 'AtomNet: A deep convolutional neural network for bioactivity prediction in structure-based drug discovery',\n",
       "   'key': 'ref26'},\n",
       "  {'doi-asserted-by': 'publisher',\n",
       "   'key': 'ref25',\n",
       "   'DOI': '10.1007/s10822-015-9893-9'}],\n",
       " 'event': '2020 International Conference on 3D Vision (3DV)',\n",
       " 'container-title': '2020 International Conference on 3D Vision (3DV)',\n",
       " 'original-title': [],\n",
       " 'link': [{'URL': 'http://xplorestaging.ieee.org/ielx7/9320095/9320098/09320391.pdf?arnumber=9320391',\n",
       "   'content-type': 'unspecified',\n",
       "   'content-version': 'vor',\n",
       "   'intended-application': 'similarity-checking'}],\n",
       " 'deposited': {'date-parts': [[2022, 6, 28]],\n",
       "  'date-time': '2022-06-28T21:48:52Z',\n",
       "  'timestamp': 1656452932000},\n",
       " 'score': 1,\n",
       " 'resource': {'primary': {'URL': 'https://ieeexplore.ieee.org/document/9320391/'}},\n",
       " 'subtitle': [],\n",
       " 'short-title': [],\n",
       " 'issued': {'date-parts': [[2020, 11]]},\n",
       " 'references-count': 28,\n",
       " 'URL': 'http://dx.doi.org/10.1109/3DV50981.2020.00103',\n",
       " 'relation': {},\n",
       " 'subject': [],\n",
       " 'published': {'date-parts': [[2020, 11]]}}"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from doiref import CrossRefClient\n",
    "\n",
    "session = CrossRefClient()\n",
    "\n",
    "ref = session.doi2json('http://dx.doi.org/10.1109/3DV50981.2020.00103')\n",
    "ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "corresponding = ['Meiler']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def formatSPP(ref):\n",
    "    s = \"\"\n",
    "    preprint = False\n",
    "    for author in ref['author']:\n",
    "        s += author['given'][0] + \". \" + author['family'] + \",\"\n",
    "        if author['family'] in corresponding:\n",
    "            s += '*'\n",
    "        s += ' '\n",
    "    s = s[:-1] + '\\n'\n",
    "    s += ref['title'] + '\\n'\n",
    "    if 'container-title-short' in ref:\n",
    "        s += ref['container-title-short'] + \" \"\n",
    "    elif 'container-title' in ref and ref['container-title']:\n",
    "        s += ref['container-title'] + \" \"\n",
    "    else:\n",
    "        if ref['publisher'] == 'Cold Spring Harbor Laboratory':\n",
    "            s += 'bioRxiv '\n",
    "            preprint = True\n",
    "        elif ref['publisher'] == 'American Chemical Society (ACS)':\n",
    "            s += 'ChemRxiv '\n",
    "            preprint = True\n",
    "        elif ref['publisher'] == 'arXiv':\n",
    "            s += 'arXiv '\n",
    "            preprint = True\n",
    "        else:\n",
    "            raise KeyError('container-title-short')\n",
    "    if 'created' in ref:\n",
    "        s += str(ref['created']['date-parts'][0][0])\n",
    "    elif 'issued' in ref:\n",
    "        s += str(ref['issued']['date-parts'][0][0])\n",
    "    else:\n",
    "        raise KeyError('year')\n",
    "    if not preprint:\n",
    "        if 'volume' in ref : s += \", \" + str(ref['volume'])\n",
    "        if 'page' in ref : s += \", \" + str(ref['page'])\n",
    "    s += \". \" + ref['URL']\n",
    "    return s, preprint\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 291/291 [03:23<00:00,  1.43it/s]\n"
     ]
    }
   ],
   "source": [
    "all_refs = {}\n",
    "\n",
    "for doi in tqdm(joint_dois):\n",
    "    all_refs[doi] = session.doi2json(doi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "68\n"
     ]
    }
   ],
   "source": [
    "ai_dois = set()\n",
    "\n",
    "buzzwords = [' AI ', ' ML ', 'intelligence', 'learning', 'training', 'alphafold', 'qsar', 'neural network', \n",
    "             'esm',\n",
    "             'computational methods in drug discovery', 'ligand-based virtual screen for the discovery of novel m5 inhibitor chemotypes',\n",
    "             'computer-aided design and biological evaluation of diazaspirocyclic']\n",
    "\n",
    "for doi in joint_dois:\n",
    "    ref = all_refs[doi]\n",
    "    title = str(ref.get('title', \"\")).casefold()\n",
    "    abstract = str(ref.get('abstract', \"\")).casefold()\n",
    "    buzzed = False\n",
    "    for bw in buzzwords:\n",
    "        if bw.casefold() in title or bw.casefold() in abstract:\n",
    "            ai_dois.add(doi)\n",
    "            buzzed = True\n",
    "            #print(title, doi, bw)\n",
    "            break\n",
    "    if not buzzed:\n",
    "        continue\n",
    "        print(doi)\n",
    "        print(title)\n",
    "        print(abstract)\n",
    "\n",
    "print(len(ai_dois))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S. Jiang, Z. Su, N. Bloodworth, Y. Liu, C. Martina, D. Harrison, J. Meiler,*\n",
      "Machine learning application to predict binding affinity between peptide containing non-canonical amino acids and HLA0201\n",
      "bioRxiv 2024. http://dx.doi.org/10.1101/2024.11.19.624425 \n",
      "\n",
      "G. Sliwoski, S. Kothiwale, J. Meiler,* E. Lowe,\n",
      "Computational Methods in Drug Discovery\n",
      "Pharmacol Rev 2014, 66, 334-395. http://dx.doi.org/10.1124/pr.112.007336 \n",
      "\n",
      "J. Leman,, R. Mueller, M. Karakas, N. Woetzel, J. Meiler,*\n",
      "Simultaneous prediction of protein secondary structure and transmembrane spans\n",
      "Proteins 2013, 81, 1127-1140. http://dx.doi.org/10.1002/prot.24258 \n",
      "\n",
      "G. Sliwoski, J. Mendenhall, J. Meiler,*\n",
      "Autocorrelation descriptor improvements for QSAR: 2DA_Sign and 3DA_Sign\n",
      "J Comput Aided Mol Des 2015, 30, 209-217. http://dx.doi.org/10.1007/s10822-015-9893-9 \n",
      "\n",
      "Y. Liu, Y. Wang, O. Vu, R. Moretti, B. Bodenheimer, J. Meiler,* T. Derr,\n",
      "Interpretable Chirality-Aware Graph Neural Network for Quantitative Structure Activity Relationship Modeling in Drug Discovery\n",
      "AAAI 2023, 37, 14356-14364. http://dx.doi.org/10.1609/aaai.v37i12.26679 \n",
      "\n",
      "A. Blee, B. Li, T. Pecen, J. Meiler,* Z. Nagel, J. Capra, W. Chazin,\n",
      "An active learning framework improves tumor variant interpretation\n",
      "bioRxiv 2021. http://dx.doi.org/10.1101/2021.11.08.467747 \n",
      "\n",
      "A. Gulsevin, B. Han, J. Porta, H. Mchaourab, J. Meiler,* A. Kenworthy,\n",
      "Template-free prediction of a new monotopic membrane protein fold and assembly by AlphaFold2\n",
      "Biophysical Journal 2022, 122, 2041-2052. http://dx.doi.org/10.1016/j.bpj.2022.11.011 \n",
      "\n",
      "D. del Alamo, L. DeSousa, R. Nair, S. Rahman, J. Meiler,* H. Mchaourab,\n",
      "Integrated AlphaFold2 and DEER investigation of the conformational dynamics of a pH-dependent APC antiporter\n",
      "Proc. Natl. Acad. Sci. U.S.A. 2022, 119. http://dx.doi.org/10.1073/pnas.2206129119 \n",
      "\n",
      "S. DeLuca, J. Meiler,*\n",
      "A Novel Method for Guiding Protein-Ligand Docking with QSAR-Derived Pharmacophore Maps\n",
      "Biophysical Journal 2011, 100, 394a-395a. http://dx.doi.org/10.1016/j.bpj.2010.12.2344 \n",
      "\n",
      "J. Koehler, N. Woetzel, R. Staritzbichler, C. Sanders, J. Meiler,*\n",
      "A unified hydrophobicity scale for multispan membrane proteins\n",
      "Proteins 2008, 76, 13-29. http://dx.doi.org/10.1002/prot.22315 \n",
      "\n",
      "Y. Liu, Y. Wang, O. Vu, R. Moretti, B. Bodenheimer, J. Meiler,* T. Derr,\n",
      "Interpretable Chirality-Aware Graph Neural Network for Quantitative Structure Activity Relationship Modeling in Drug Discovery\n",
      "bioRxiv 2022. http://dx.doi.org/10.1101/2022.08.24.505155 \n",
      "\n",
      "B. Brown, R. Stein, J. Meiler,* H. Mchaourab,\n",
      "Approximating conformational Boltzmann distributions with AlphaFold2 predictions\n",
      "bioRxiv 2023. http://dx.doi.org/10.1101/2023.08.06.552168 \n",
      "\n",
      "J. Meiler,* R. Meusinger, M. Will,\n",
      "Fast Determination of <sup>13</sup>C NMR Chemical Shifts Using Artificial Neural Networks\n",
      "J. Chem. Inf. Comput. Sci. 2002, 40, 1169-1176. http://dx.doi.org/10.1021/ci000021c \n",
      "\n",
      "O. Vu, J. Mendenhall, D. Altarawy, J. Meiler,*\n",
      "BCL::Mol2D—a robust atom environment descriptor for QSAR modeling and lead optimization\n",
      "J Comput Aided Mol Des 2019, 33, 477-486. http://dx.doi.org/10.1007/s10822-019-00199-8 \n",
      "\n",
      "B. Li, J. Mendenhall, J. Capra, J. Meiler,*\n",
      "A Multitask Deep-Learning Method for Predicting Membrane Associations and Secondary Structures of Proteins\n",
      "J. Proteome Res. 2021, 20, 4089-4100. http://dx.doi.org/10.1021/acs.jproteome.1c00410 \n",
      "\n",
      "G. Sliwoski, E. Lowe, M. Butkiewicz, J. Meiler,*\n",
      "BCL::EMAS — Enantioselective Molecular Asymmetry Descriptor for 3D-QSAR\n",
      "Molecules 2012, 17, 9971-9989. http://dx.doi.org/10.3390/molecules17089971 \n",
      "\n",
      "D. Sala, F. Engelberger, H. Mchaourab, J. Meiler,*\n",
      "Modeling conformational states of proteins with AlphaFold\n",
      "Current Opinion in Structural Biology 2023, 81, 102645. http://dx.doi.org/10.1016/j.sbi.2023.102645 \n",
      "\n",
      "E. Lowe, M. Butkiewicz, M. Spellings, A. Omlor, J. Meiler,*\n",
      "Comparative analysis of machine learning techniques for the prediction of logP\n",
      "2011 IEEE Symposium on Computational Intelligence in Bioinformatics and Computational Biology (CIBCB) 2011, 1-6. http://dx.doi.org/10.1109/CIBCB.2011.5948478 \n",
      "\n",
      "A. Blee, B. Li, T. Pecen, J. Meiler,* Z. Nagel, J. Capra, W. Chazin,\n",
      "An Active Learning Framework Improves Tumor Variant Interpretation\n",
      "Cancer Research 2022, 82, 2704-2715. http://dx.doi.org/10.1158/0008-5472.CAN-21-3798 \n",
      "\n",
      "M. Kuhn, J. Meiler,* D. Baker,\n",
      "Strand‐loop‐strand motifs: Prediction of hairpins and diverging turns in proteins\n",
      "Proteins 2003, 54, 282-288. http://dx.doi.org/10.1002/prot.10589 \n",
      "\n",
      "L. Somermeyer, A. Fleiss, A. Mishin, N. Bozhanova, A. Igolkina, J. Meiler,* M. Pujol, E. Putintseva, K. Sarkisyan, F. Kondrashov,\n",
      "Heterogeneity of the GFP fitness landscape and data-driven protein design\n",
      "bioRxiv 2021. http://dx.doi.org/10.1101/2021.12.08.471728 \n",
      "\n",
      "E. Lowe, M. Butkiewicz, N. Woetzel, J. Meiler,*\n",
      "GPU-accelerated machine learning techniques enable QSAR modeling of large HTS data\n",
      "2012 IEEE Symposium on Computational Intelligence in Bioinformatics and Computational Biology (CIBCB) 2012, 314-320. http://dx.doi.org/10.1109/CIBCB.2012.6217246 \n",
      "\n",
      "J. Meiler,* M. Köck,\n",
      "Novel methods of automated structure elucidation based on <sup>13</sup>C NMR spectroscopy\n",
      "Magnetic Reson in Chemistry 2004, 42, 1042-1045. http://dx.doi.org/10.1002/mrc.1424 \n",
      "\n",
      "Y. Liu, R. Moretti, Y. Wang, H. Dong, B. Yan, B. Bodenheimer, T. Derr, J. Meiler,*\n",
      "Advancements in Ligand-Based Virtual Screening through the Synergistic Integration of Graph Neural Networks and Expert-Crafted Descriptors\n",
      "bioRxiv 2023. http://dx.doi.org/10.1101/2023.04.17.537185 \n",
      "\n",
      "J. Mendenhall, J. Meiler,*\n",
      "Improving quantitative structure–activity relationship models using Artificial Neural Networks trained with dropout\n",
      "J Comput Aided Mol Des 2016, 30, 177-189. http://dx.doi.org/10.1007/s10822-016-9895-2 \n",
      "\n",
      "E. Lowe, A. Ferrebee, A. Rodriguez, P. Conn, J. Meiler,*\n",
      "3D-QSAR CoMFA study of benzoxazepine derivatives as mGluR5 positive allosteric modulators\n",
      "Bioorganic &amp; Medicinal Chemistry Letters 2010, 20, 5922-5924. http://dx.doi.org/10.1016/j.bmcl.2010.07.061 \n",
      "\n",
      "V. Golkov, M. Skwark, A. Mirchev, G. Dikov, A. Geanes, J. Mendenhall, J. Meiler,* D. Cremers,\n",
      "3D Deep Learning for Biological Function Prediction from Physical Fields\n",
      "2020 International Conference on 3D Vision (3DV) 2021, 928-937. http://dx.doi.org/10.1109/3DV50981.2020.00103 \n",
      "\n",
      "M. Butkiewicz, R. Mueller, D. Selic, E. Dawson, J. Meiler,*\n",
      "Application of machine learning approaches on quantitative structure activity relationships\n",
      "2009 IEEE Symposium on Computational Intelligence in Bioinformatics and Computational Biology 2009, 255-262. http://dx.doi.org/10.1109/CIBCB.2009.4925736 \n",
      "\n",
      "V. Golkov, A. Becker, D. Plop, D. Čuturilo, N. Davoudi, J. Mendenhall, R. Moretti, J. Meiler,* D. Cremers,\n",
      "Deep Learning for Virtual Screening: Five Reasons to Use ROC Cost Functions\n",
      "arXiv 2020. https://arxiv.org/abs/2007.07029 \n",
      "\n",
      "S. Phul, G. Kuenze, C. Vanoye, C. Sanders, A. George, J. Meiler,*\n",
      "Predicting the Functional Impact of KCNQ1 Variants with Artificial Neural Networks\n",
      "bioRxiv 2021. http://dx.doi.org/10.1101/2021.12.06.471347 \n",
      "\n",
      "S. Phul, G. Kuenze, C. Vanoye, C. Sanders, A. George, J. Meiler,*\n",
      "Predicting the functional impact of KCNQ1 variants with artificial neural networks\n",
      "PLoS Comput Biol 2022, 18, e1010038. http://dx.doi.org/10.1371/journal.pcbi.1010038 \n",
      "\n",
      "D. Alamo, L. DeSousa, R. Nair, S. Rahman, J. Meiler,* H. Mchaourab,\n",
      "Integrated AlphaFold2 and DEER investigation of the conformational dynamics of a pH-dependent APC antiporter\n",
      "bioRxiv 2022. http://dx.doi.org/10.1101/2022.04.13.488180 \n",
      "\n",
      "L. Gonzalez Somermeyer, A. Fleiss, A. Mishin, N. Bozhanova, A. Igolkina, J. Meiler,* M. Alaball Pujol, E. Putintseva, K. Sarkisyan, F. Kondrashov,\n",
      "Heterogeneity of the GFP fitness landscape and data-driven protein design\n",
      "eLife 2022, 11. http://dx.doi.org/10.7554/eLife.75842 \n",
      "\n",
      "R. Mueller, E. Dawson, J. Meiler,* A. Rodriguez, B. Chauder, B. Bates, A. Felts, J. Lamb, U. Menon, S. Jadhav, A. Kane, C. Jones, K. Gregory, C. Niswender, P. Conn, C. Olsen, D. Winder, K. Emmitte, C. Lindsley,\n",
      "Discovery of 2‐(2‐Benzoxazoyl amino)‐4‐Aryl‐5‐Cyanopyrimidine as Negative Allosteric Modulators (NAMs) of Metabotropic Glutamate Receptor 5 (mGlu<sub>5</sub>): From an Artificial Neural Network Virtual Screen to an In Vivo Tool Compound\n",
      "ChemMedChem 2012, 7, 406-414. http://dx.doi.org/10.1002/cmdc.201100510 \n",
      "\n",
      "V. Golkov, A. Becker, D. Plop, D. Čuturilo, N. Davoudi, J. Mendenhall, R. Moretti, J. Meiler,* D. Cremers,\n",
      "Deep Learning for Virtual Screening: Five Reasons to Use ROC Cost Functions\n",
      "bioRxiv 2020. http://dx.doi.org/10.1101/2020.06.25.166884 \n",
      "\n",
      "M. Ertelt, J. Meiler,* C. Schoeder,\n",
      "Combining Rosetta Sequence Design with Protein Language Model Predictions Using Evolutionary Scale Modeling (ESM) as Restraint\n",
      "ACS Synth. Biol. 2024, 13, 1085-1092. http://dx.doi.org/10.1021/acssynbio.3c00753 \n",
      "\n",
      "M. Fischer, J. Crowe, J. Meiler,*\n",
      "Computational epitope mapping of class I fusion proteins using low complexity supervised learning methods\n",
      "PLoS Comput Biol 2022, 18, e1010230. http://dx.doi.org/10.1371/journal.pcbi.1010230 \n",
      "\n",
      "C. Kuntz, H. Woods, A. McKee, N. Zelt, J. Mendenhall, J. Meiler,* J. Schlebach,\n",
      "Towards Generalizable Predictions for the Effects of Mutations on G-Protein Coupled Receptor Expression\n",
      "bioRxiv 2021. http://dx.doi.org/10.1101/2021.12.28.474371 \n",
      "\n",
      "X. Kuang, z. su, Y. Liu, X. Lin, J. Smith, T. Derr, Y. Wu, J. Meiler,*\n",
      "SuperWater: Predicting Water Molecule Positions on Protein Structures by Generative AI\n",
      "bioRxiv 2024. http://dx.doi.org/10.1101/2024.11.18.624208 \n",
      "\n",
      "A. Geanes, H. Cho, K. Nance, K. McGowan, P. Conn, C. Jones, J. Meiler,* C. Lindsley,\n",
      "Ligand-based virtual screen for the discovery of novel M5 inhibitor chemotypes\n",
      "Bioorganic &amp; Medicinal Chemistry Letters 2016, 26, 4487-4491. http://dx.doi.org/10.1016/j.bmcl.2016.07.071 \n",
      "\n",
      "J. Meiler,* D. Baker,\n",
      "Coupled prediction of protein secondary and tertiary structure\n",
      "Proc. Natl. Acad. Sci. U.S.A. 2003, 100, 12105-12110. http://dx.doi.org/10.1073/pnas.1831973100 \n",
      "\n",
      "E. McDonald, K. Oliver, J. Schlebach, J. Meiler,* L. Plate,\n",
      "Benchmarking AlphaMissense pathogenicity predictions against cystic fibrosis variants\n",
      "PLoS ONE 2024, 19, e0297560. http://dx.doi.org/10.1371/journal.pone.0297560 \n",
      "\n",
      "J. DeCorte, B. Brown, J. Meiler,*\n",
      "Interpretable deep-learning pKa prediction for small molecule drugs via atomic sensitivity analysis\n",
      "ChemRxiv 2024. http://dx.doi.org/10.26434/chemrxiv-2024-hr692 \n",
      "\n",
      "D. Sala, P. Hildebrand, J. Meiler,*\n",
      "Biasing AlphaFold2 to predict GPCRs and kinases with user-defined functional or structural properties\n",
      "Front. Mol. Biosci. 2023, 10. http://dx.doi.org/10.3389/fmolb.2023.1121962 \n",
      "\n",
      "J. Meiler,* M. Will,\n",
      "Automated Structure Elucidation of Organic Molecules from <sup>13</sup>C NMR Spectra Using Genetic Algorithms and Neural Networks\n",
      "J. Chem. Inf. Comput. Sci. 2002, 41, 1535-1546. http://dx.doi.org/10.1021/ci0102970 \n",
      "\n",
      "J. Meiler,* R. Meusinger, M. Will,\n",
      "Neural Network Prediction of 13C NMR Chemical Shifts of Substituted Benzenes\n",
      "Monatshefte fuer Chemie 2006, 130, 1089-1095. http://dx.doi.org/10.1007/PL00010287 \n",
      "\n",
      "D. Sala, J. Meiler,*\n",
      "Biasing AlphaFold2 to predict GPCRs and Kinases with user-defined functional or structural properties\n",
      "bioRxiv 2022. http://dx.doi.org/10.1101/2022.12.11.519936 \n",
      "\n",
      "E. McDonald, K. Oliver, J. Schlebach, J. Meiler,* L. Plate,\n",
      "Benchmarking AlphaMissense Pathogenicity Predictions Against Cystic Fibrosis Variants\n",
      "bioRxiv 2023. http://dx.doi.org/10.1101/2023.10.05.561147 \n",
      "\n",
      "A. Gulsevin, J. Meiler,*\n",
      "Benchmarking Peptide Structure Prediction with AlphaFold2\n",
      "bioRxiv 2022. http://dx.doi.org/10.1101/2022.02.17.480937 \n",
      "\n",
      "M. Ertelt, R. Moretti, J. Meiler,* C. Schoeder,\n",
      "Self-supervised machine learning methods for protein design improve sampling, but not the identification of high-fitness variants\n",
      "bioRxiv 2024. http://dx.doi.org/10.1101/2024.06.20.599843 \n",
      "\n",
      "D. del Alamo, D. Sala, H. Mchaourab, J. Meiler,*\n",
      "Sampling alternative conformational states of transporters and receptors with AlphaFold2\n",
      "eLife 2022, 11. http://dx.doi.org/10.7554/eLife.75751 \n",
      "\n",
      "A. Gulsevin, B. Han, J. Porta, H. Mchaourab, J. Meiler,* A. Kenworthy,\n",
      "Template-free prediction of a new monotopic membrane protein fold and oligomeric assembly by Alphafold2\n",
      "bioRxiv 2022. http://dx.doi.org/10.1101/2022.07.12.499809 \n",
      "\n",
      "B. Brown, R. Stein, J. Meiler,* H. Mchaourab,\n",
      "Approximating Projections of Conformational Boltzmann Distributions with AlphaFold2 Predictions: Opportunities and Limitations\n",
      "J. Chem. Theory Comput. 2024, 20, 1434-1447. http://dx.doi.org/10.1021/acs.jctc.3c01081 \n",
      "\n",
      "B. Brown, O. Vu, A. Geanes, S. Kothiwale, M. Butkiewicz, E. Lowe, R. Mueller, R. Pape, J. Mendenhall, J. Meiler,*\n",
      "Introduction to the BioChemical Library (BCL): An Application-Based Open-Source Toolkit for Integrated Cheminformatics and Machine Learning in Computer-Aided Drug Discovery\n",
      "Front. Pharmacol. 2022, 13. http://dx.doi.org/10.3389/fphar.2022.833099 \n",
      "\n",
      "S. Kothiwale, C. Borza, A. Pozzi, J. Meiler,*\n",
      "Quantitative Structure–Activity Relationship Modeling of Kinase Selectivity Profiles\n",
      "Molecules 2017, 22, 1576. http://dx.doi.org/10.3390/molecules22091576 \n",
      "\n",
      "B. Brown, J. Mendenhall, A. Geanes, J. Meiler,*\n",
      "General Purpose Structure-Based Drug Discovery Neural Network Score Functions with Human-Interpretable Pharmacophore Maps\n",
      "J. Chem. Inf. Model. 2021, 61, 603-620. http://dx.doi.org/10.1021/acs.jcim.0c01001 \n",
      "\n",
      "C. Moth, J. Sheehan, A. Mamun, R. Sivley, A. Gulsevin, D. Rinker, J. Capra, J. Meiler,*\n",
      "VUStruct: a compute pipeline for high throughput and personalized structural biology\n",
      "bioRxiv 2024. http://dx.doi.org/10.1101/2024.08.06.606224 \n",
      "\n",
      "M. Butkiewicz, E. Lowe, R. Mueller, J. Mendenhall, P. Teixeira, C. Weaver, J. Meiler,*\n",
      "Benchmarking Ligand-Based Virtual High-Throughput Screening with the PubChem Database\n",
      "Molecules 2013, 18, 735-756. http://dx.doi.org/10.3390/molecules18010735 \n",
      "\n",
      "J. Koehler, R. Mueller, J. Meiler,*\n",
      "Improved prediction of trans-membrane spans in proteins using an artificial neural network\n",
      "2009 IEEE Symposium on Computational Intelligence in Bioinformatics and Computational Biology 2009, 68-74. http://dx.doi.org/10.1109/CIBCB.2009.4925709 \n",
      "\n",
      "C. Jones, B. Brown, D. Schultz, J. Engers, V. Kramlinger, J. Meiler,* C. Lindsley,\n",
      "Computer-Aided Design and Biological Evaluation of Diazaspirocyclic D<sub>4</sub>R Antagonists\n",
      "ACS Chem. Neurosci. 2024, 15, 2396-2407. http://dx.doi.org/10.1021/acschemneuro.4c00086 \n",
      "\n",
      "E. McDonald, T. Jones, L. Plate, J. Meiler,* A. Gulsevin,\n",
      "Benchmarking AlphaFold2 on peptide structure prediction\n",
      "Structure 2022, 31, 111-119.e2. http://dx.doi.org/10.1016/j.str.2022.11.012 \n",
      "\n",
      "D. del Alamo, D. Sala, H. Mchaourab, J. Meiler,*\n",
      "Sampling the conformational landscapes of transporters and receptors with AlphaFold2\n",
      "bioRxiv 2021. http://dx.doi.org/10.1101/2021.11.22.469536 \n",
      "\n",
      "M. Butkiewicz, E. Lowe, J. Meiler,*\n",
      "Bcl&amp;#x2237;ChemInfo - Qualitative analysis of machine learning models for activation of HSD involved in Alzheimer's Disease\n",
      "2012 IEEE Symposium on Computational Intelligence in Bioinformatics and Computational Biology (CIBCB) 2012, 329-334. http://dx.doi.org/10.1109/CIBCB.2012.6217248 \n",
      "\n",
      "M. Ertelt, V. Mulligan, J. Maguire, S. Lyskov, R. Moretti, T. Schiffner, J. Meiler,* C. Schoeder,\n",
      "Combining machine learning with structure-based protein design to predict and engineer post-translational modifications of proteins\n",
      "PLoS Comput Biol 2024, 20, e1011939. http://dx.doi.org/10.1371/journal.pcbi.1011939 \n",
      "\n",
      "A. Bleckmann, J. Meiler,*\n",
      "Epothilones: Quantitative Structure Activity Relations Studied by Support Vector Machines and Artificial Neural Networks\n",
      "QSAR Comb. Sci. 2003, 22, 722-728. http://dx.doi.org/10.1002/qsar.200330837 \n",
      "\n",
      "J. Meiler,* W. Maier, M. Will, R. Meusinger,\n",
      "Using Neural Networks for 13C NMR Chemical Shift Prediction–Comparison with Traditional Methods\n",
      "Journal of Magnetic Resonance 2002, 157, 242-252. http://dx.doi.org/10.1006/jmre.2002.2599 \n",
      "\n",
      "[['given', 'https://arxiv.org/abs/2411.09820', 'arXiv'], ['given', 'http://dx.doi.org/10.7554/eLife.92406.1', 'eLife Sciences Publications, Ltd']]\n"
     ]
    }
   ],
   "source": [
    "errors = []\n",
    "\n",
    "for doi in list(ai_dois):\n",
    "    ref = all_refs[doi]\n",
    "    try:\n",
    "        #if 'created' in ref:\n",
    "        #    if ref['created']['date-parts'][0][0] < 2022:\n",
    "        #        continue\n",
    "        #elif 'issued' in ref:\n",
    "        #    if ref['issued']['date-parts'][0][0] < 2022:\n",
    "        #        continue\n",
    "        citation, preprint = formatSPP(ref)\n",
    "        #if not preprint:\n",
    "        #    print(citation, '\\n')\n",
    "        print(citation, '\\n')\n",
    "    except KeyError as e:\n",
    "        errors.append([e.args[0], ref['URL'], ref['publisher']])\n",
    "print(errors)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "publication_list",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
