{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from scholarly import scholarly\n",
    "\n",
    "search_query = scholarly.search_author('Jens Meiler')\n",
    "#author = next(search_query, None)\n",
    "#scholarly.fill(author)\n",
    "\n",
    "# not working because google blocked me"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from tqdm import tqdm\n",
    "import requests\n",
    "\n",
    "jens_orcid = '0000-0001-8945-193X'\n",
    "\n",
    "orcid_api_url = f\"https://pub.orcid.org/v3.0/{jens_orcid}/works\"\n",
    "headers = {\"Accept\": \"application/json\"}\n",
    "\n",
    "response = requests.get(orcid_api_url, headers=headers)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "267 of 298\n"
     ]
    }
   ],
   "source": [
    "data = response.json()\n",
    "works = data.get('group', [])\n",
    "\n",
    "auto_dois = []\n",
    "\n",
    "for work in works:\n",
    "        work_summary = work.get('work-summary', [])\n",
    "        for summary in work_summary:\n",
    "            if 'external-ids' in summary:\n",
    "                external_ids = summary['external-ids']['external-id']\n",
    "                for external_id in external_ids:\n",
    "                    if external_id.get('external-id-type') == 'doi':\n",
    "                        auto_dois.append('https://doi.org/' + external_id.get('external-id-value'))\n",
    "\n",
    "print(len(auto_dois), 'of', len(works))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "manual_dois = \"\"\"https://doi.org/10.1101/2024.08.06.606224 \n",
    "\n",
    "https://doi.org/10.48550/arXiv.2411.09820 \n",
    "\n",
    "https://doi.org/10.1101/2024.06.20.599843 \n",
    "\n",
    "https://doi.org/10.26434/chemrxiv-2024-hr692 \n",
    "\n",
    "https://doi.org/10.1021/acschemneuro.4c00086 \n",
    "\n",
    "https://doi.org/10.1021/acssynbio.3c00753 \n",
    "\n",
    "https://doi.org/10.1371/journal.pcbi.1011939 \n",
    "\n",
    "https://doi.org/10.1021/acs.jctc.3c01081 \n",
    "\n",
    "https://doi.org/10.1101/2022.11.15.516482  \n",
    "\n",
    "https://doi.org/10.1016/j.sbi.2023.102645\n",
    "\n",
    "https://doi.org/10.1609/aaai.v37i12.26679 \n",
    "\n",
    "https://doi.org/10.1016/j.bpj.2022.11.011 \n",
    "\n",
    "https://doi.org/10.1101/2023.04.17.537185 \n",
    "\n",
    "https://doi.org/10.3389/fmolb.2023.1121962 \n",
    "\n",
    "https://doi.org/10.1016/j.str.2022.11.012 \n",
    "\n",
    "https://doi.org/10.1073/pnas.2206129119 \n",
    "\n",
    "https://doi.org/10.1158/0008-5472.CAN-21-3798 \n",
    "\n",
    "https://doi.org/10.7554/eLife.75751\n",
    "\n",
    "https://doi.org/10.3389/fphar.2022.833099 \n",
    "\n",
    "https://doi.org/10.1021/acs.jproteome.1c00410 \n",
    "\n",
    "https://doi.org/10.1021/acs.jcim.0c01001 \n",
    "\n",
    "https://doi.org/10.1021/acs.jcim.0c01001 \n",
    "\n",
    "https://doi.org/10.1109/3DV50981.2020.00103\n",
    "\n",
    "https://doi.org/10.48550/arXiv.2007.07029 \n",
    "\n",
    "https://doi.org/10.1007/s10822-019-00199-8 \n",
    "\n",
    "https://doi.org/10.3390/molecules22091576 \n",
    "\n",
    "https://doi.org/10.1016/j.bmcl.2016.07.071\n",
    "\n",
    "https://doi.org/10.1007/s10822-016-9895-2 \n",
    "\n",
    "https://doi.org/10.1007/s10822-015-9893-9 \n",
    "\n",
    "https://doi.org/10.1124/pr.112.007336 \n",
    "\n",
    "https://doi.org/10.3390/molecules18010735 \n",
    "\n",
    "https://doi.org/10.3390/molecules17089971 \n",
    "\n",
    "https://doi.org/10.1109/CIBCB.2012.6217248\n",
    "\n",
    "https://doi.org/10.1109/CIBCB.2012.6217246\n",
    "\n",
    "https://doi.org/10.1109/CIBCB.2011.5948478\n",
    "\n",
    "https://doi.org/10.1002/cmdc.201100510 \n",
    "\n",
    "https://doi.org/10.1109/CIBCB.2011.5948478\n",
    "\n",
    "https://doi.org/10.1016/j.bpj.2010.12.2344 \n",
    "\n",
    "https://doi.org/10.1016/j.bmcl.2010.07.061\n",
    "\n",
    "https://doi.org/10.1109/CIBCB.2009.4925736\n",
    "\n",
    "https://doi.org/10.1109/CIBCB.2009.4925709\n",
    "\n",
    "https://doi.org/10.1002/qsar.200330837 \n",
    "\n",
    "https://doi.org/10.1023/A:1023060720156 \n",
    "\n",
    "https://doi.org/10.1006/jmre.2002.2599\n",
    "\n",
    "https://doi.org/10.1021/ci0102970 \n",
    "\n",
    "https://doi.org/10.1021/ci000021c\n",
    "\n",
    "https://doi.org/10.1007/PL00010287\"\"\"\n",
    "\n",
    "manual_dois = list(set(manual_dois.split()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "45"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(manual_dois)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "291 16\n"
     ]
    }
   ],
   "source": [
    "joint_dois = set(manual_dois)\n",
    "joint_dois.update(auto_dois)\n",
    "joint_dois = list(joint_dois)\n",
    "intersect_dois = list(set(manual_dois) & set(auto_dois))\n",
    "print(len(joint_dois), len(intersect_dois))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "' @article{Golkov_2020, title={3D Deep Learning for Biological Function Prediction from Physical Fields}, url={http://dx.doi.org/10.1109/3DV50981.2020.00103}, DOI={10.1109/3dv50981.2020.00103}, journal={2020 International Conference on 3D Vision (3DV)}, publisher={IEEE}, author={Golkov, Vladimir and Skwark, Marcin J. and Mirchev, Atanas and Dikov, Georgi and Geanes, Alexander R. and Mendenhall, Jeffrey and Meiler, Jens and Cremers, Daniel}, year={2020}, month=nov, pages={928Ã¢\\x80\\x93937} }\\n'"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from doiref import CrossRefClient\n",
    "\n",
    "session = CrossRefClient()\n",
    "\n",
    "ref = session.doi2bibtex('http://dx.doi.org/10.1109/3DV50981.2020.00103')\n",
    "ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "corresponding = ['Meiler']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def formatSPP(ref):\n",
    "    s = \"\"\n",
    "    preprint = False\n",
    "    for author in ref['author']:\n",
    "        s += author['given'][0] + \". \" + author['family'] + \",\"\n",
    "        if author['family'] in corresponding:\n",
    "            s += '*'\n",
    "        s += ' '\n",
    "    s = s[:-1] + '\\n'\n",
    "    s += ref['title'] + '\\n'\n",
    "    if 'container-title-short' in ref:\n",
    "        s += ref['container-title-short'] + \" \"\n",
    "    elif 'container-title' in ref and ref['container-title']:\n",
    "        s += ref['container-title'] + \" \"\n",
    "    else:\n",
    "        if ref['publisher'] == 'Cold Spring Harbor Laboratory':\n",
    "            s += 'bioRxiv '\n",
    "            preprint = True\n",
    "        elif ref['publisher'] == 'American Chemical Society (ACS)':\n",
    "            s += 'ChemRxiv '\n",
    "            preprint = True\n",
    "        elif ref['publisher'] == 'arXiv':\n",
    "            s += 'arXiv '\n",
    "            preprint = True\n",
    "        else:\n",
    "            raise KeyError('container-title-short')\n",
    "    if 'created' in ref:\n",
    "        s += str(ref['created']['date-parts'][0][0])\n",
    "    elif 'issued' in ref:\n",
    "        s += str(ref['issued']['date-parts'][0][0])\n",
    "    else:\n",
    "        raise KeyError('year')\n",
    "    if not preprint:\n",
    "        if 'volume' in ref : s += \", \" + str(ref['volume'])\n",
    "        if 'page' in ref : s += \", \" + str(ref['page'])\n",
    "    s += \". \" + ref['URL']\n",
    "    return s, preprint\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle\n",
    "\n",
    "all_refs = {}\n",
    "\n",
    "with open('all_refs.pickle', 'rb') as file:\n",
    "    all_refs = pickle.load(file)\n",
    "\n",
    "#for doi in tqdm(joint_dois):\n",
    "#    all_refs[doi] = session.doi2json(doi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('all_refs.pickle', 'wb') as file:\n",
    "    pickle.dump(all_refs, file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "45"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "manual_titles = []\n",
    "for doi in manual_dois:\n",
    "    ref = all_refs[doi]\n",
    "    manual_titles.append(str(ref['title']).casefold())\n",
    "len(manual_titles)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "predicting the functional impact of kcnq1 variants with artificial neural networks https://doi.org/10.1371/journal.pcbi.1010038  AI \n",
      "machine learning application to predict binding affinity between peptide containing non-canonical amino acids and hla0201 https://doi.org/10.1101/2024.11.19.624425 learning\n",
      "template-free prediction of a new monotopic membrane protein fold and oligomeric assembly by alphafold2 https://doi.org/10.1101/2022.07.12.499809 learning\n",
      "divergent pairwise epistasis in the context of unstable membrane protein variants https://doi.org/10.7554/eLife.92406.1 learning\n",
      "heterogeneity of the gfp fitness landscape and data-driven protein design https://doi.org/10.1101/2021.12.08.471728 learning\n",
      "computational epitope mapping of class i fusion proteins using low complexity supervised learning methods https://doi.org/10.1371/journal.pcbi.1010230 learning\n",
      "approximating conformational boltzmann distributions with alphafold2 predictions https://doi.org/10.1101/2023.08.06.552168 alphafold\n",
      "predicting the functional impact of kcnq1 variants with artificial neural networks https://doi.org/10.1101/2021.12.06.471347  AI \n",
      "heterogeneity of the gfp fitness landscape and data-driven protein design https://doi.org/10.7554/eLife.75842 learning\n",
      "simultaneous prediction of protein secondary structure and transmembrane spans https://doi.org/10.1002/prot.24258 neural network\n",
      "benchmarking alphamissense pathogenicity predictions against cystic fibrosis variants https://doi.org/10.1101/2023.10.05.561147 training\n",
      "coupled prediction of protein secondary and tertiary structure https://doi.org/10.1073/pnas.1831973100 neural network\n",
      "strandâloopâstrand motifs: prediction of hairpins and diverging turns in proteins https://doi.org/10.1002/prot.10589 neural network\n",
      "superwater: predicting water molecule positions on protein structures by generative ai https://doi.org/10.1101/2024.11.18.624208  AI \n",
      "benchmarking peptide structure prediction with alphafold2 https://doi.org/10.1101/2022.02.17.480937 alphafold\n",
      "sampling the conformational landscapes of transporters and receptors with alphafold2 https://doi.org/10.1101/2021.11.22.469536 learning\n",
      "a unified hydrophobicity scale for multispan membrane proteins https://doi.org/10.1002/prot.22315 learning\n",
      "towards generalizable predictions for the effects of mutations on g-protein coupled receptor expression https://doi.org/10.1101/2021.12.28.474371 neural network\n",
      "novel methods of automated structure elucidation based on <sup>13</sup>c nmr spectroscopy https://doi.org/10.1002/mrc.1424 neural network\n",
      "benchmarking alphamissense pathogenicity predictions against cystic fibrosis variants https://doi.org/10.1371/journal.pone.0297560 training\n",
      "63\n"
     ]
    }
   ],
   "source": [
    "ai_dois = set()\n",
    "\n",
    "buzzwords = [' AI ', ' ML ', 'intelligence', 'learning', 'training', 'alphafold', 'qsar', 'neural network', \n",
    "             'esm',\n",
    "             'computational methods in drug discovery', 'ligand-based virtual screen for the discovery of novel m5 inhibitor chemotypes',\n",
    "             'computer-aided design and biological evaluation of diazaspirocyclic']\n",
    "\n",
    "blacklist = ['']\n",
    "\n",
    "for doi in joint_dois:\n",
    "    ref = all_refs[doi]\n",
    "    title = str(ref.get('title', \"\")).casefold()\n",
    "    abstract = str(ref.get('abstract', \"\")).casefold()\n",
    "    buzzed = False\n",
    "    for bw in buzzwords:\n",
    "        if bw.casefold() in title or bw.casefold() in abstract:\n",
    "            buzzed = True\n",
    "            if doi in manual_dois:\n",
    "                ai_dois.add(doi)\n",
    "            else:\n",
    "                if title not in manual_titles:\n",
    "                    ai_dois.add(doi)\n",
    "                    print(title, doi, bw)\n",
    "            break\n",
    "    if not buzzed:\n",
    "        continue\n",
    "        print(doi)\n",
    "        print(title)\n",
    "        print(abstract)\n",
    "\n",
    "print(len(ai_dois))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S. Kothiwale, C. Borza, A. Pozzi, J. Meiler,*\n",
      "Quantitative StructureâActivity Relationship Modeling of Kinase Selectivity Profiles\n",
      "Molecules 2017, 22, 1576. http://dx.doi.org/10.3390/molecules22091576 \n",
      "\n",
      "Y. Liu, R. Moretti, Y. Wang, H. Dong, B. Yan, B. Bodenheimer, T. Derr, J. Meiler,*\n",
      "Advancements in Ligand-Based Virtual Screening through the Synergistic Integration of Graph Neural Networks and Expert-Crafted Descriptors\n",
      "bioRxiv 2023. http://dx.doi.org/10.1101/2023.04.17.537185 \n",
      "\n",
      "J. DeCorte, B. Brown, J. Meiler,*\n",
      "Interpretable deep-learning pKa prediction for small molecule drugs via atomic sensitivity analysis\n",
      "ChemRxiv 2024. http://dx.doi.org/10.26434/chemrxiv-2024-hr692 \n",
      "\n",
      "S. Phul, G. Kuenze, C. Vanoye, C. Sanders, A. George, J. Meiler,*\n",
      "Predicting the functional impact of KCNQ1 variants with artificial neural networks\n",
      "PLoS Comput Biol 2022, 18, e1010038. http://dx.doi.org/10.1371/journal.pcbi.1010038 \n",
      "\n",
      "D. Sala, P. Hildebrand, J. Meiler,*\n",
      "Biasing AlphaFold2 to predict GPCRs and kinases with user-defined functional or structural properties\n",
      "Front. Mol. Biosci. 2023, 10. http://dx.doi.org/10.3389/fmolb.2023.1121962 \n",
      "\n",
      "A. Gulsevin, B. Han, J. Porta, H. Mchaourab, J. Meiler,* A. Kenworthy,\n",
      "Template-free prediction of a new monotopic membrane protein fold and assembly by AlphaFold2\n",
      "Biophysical Journal 2022, 122, 2041-2052. http://dx.doi.org/10.1016/j.bpj.2022.11.011 \n",
      "\n",
      "J. Meiler,* D. Baker,\n",
      "Coupled prediction of protein secondary and tertiary structure\n",
      "Proc. Natl. Acad. Sci. U.S.A. 2003, 100, 12105-12110. http://dx.doi.org/10.1073/pnas.1831973100 \n",
      "\n",
      "G. Sliwoski, J. Mendenhall, J. Meiler,*\n",
      "Autocorrelation descriptor improvements for QSAR: 2DA_Sign and 3DA_Sign\n",
      "J Comput Aided Mol Des 2015, 30, 209-217. http://dx.doi.org/10.1007/s10822-015-9893-9 \n",
      "\n",
      "J. Meiler,* M. Will,\n",
      "Automated Structure Elucidation of Organic Molecules from <sup>13</sup>C NMR Spectra Using Genetic Algorithms and Neural Networks\n",
      "J. Chem. Inf. Comput. Sci. 2002, 41, 1535-1546. http://dx.doi.org/10.1021/ci0102970 \n",
      "\n",
      "X. Kuang, z. su, Y. Liu, X. Lin, J. Smith, T. Derr, Y. Wu, J. Meiler,*\n",
      "SuperWater: Predicting Water Molecule Positions on Protein Structures by Generative AI\n",
      "bioRxiv 2024. http://dx.doi.org/10.1101/2024.11.18.624208 \n",
      "\n",
      "G. Sliwoski, E. Lowe, M. Butkiewicz, J. Meiler,*\n",
      "BCL::EMAS â Enantioselective Molecular Asymmetry Descriptor for 3D-QSAR\n",
      "Molecules 2012, 17, 9971-9989. http://dx.doi.org/10.3390/molecules17089971 \n",
      "\n",
      "E. Lowe, M. Butkiewicz, N. Woetzel, J. Meiler,*\n",
      "GPU-accelerated machine learning techniques enable QSAR modeling of large HTS data\n",
      "2012 IEEE Symposium on Computational Intelligence in Bioinformatics and Computational Biology (CIBCB) 2012, 314-320. http://dx.doi.org/10.1109/CIBCB.2012.6217246 \n",
      "\n",
      "A. Blee, B. Li, T. Pecen, J. Meiler,* Z. Nagel, J. Capra, W. Chazin,\n",
      "An Active Learning Framework Improves Tumor Variant Interpretation\n",
      "Cancer Research 2022, 82, 2704-2715. http://dx.doi.org/10.1158/0008-5472.CAN-21-3798 \n",
      "\n",
      "S. Jiang, Z. Su, N. Bloodworth, Y. Liu, C. Martina, D. Harrison, J. Meiler,*\n",
      "Machine learning application to predict binding affinity between peptide containing non-canonical amino acids and HLA0201\n",
      "bioRxiv 2024. http://dx.doi.org/10.1101/2024.11.19.624425 \n",
      "\n",
      "J. Koehler, R. Mueller, J. Meiler,*\n",
      "Improved prediction of trans-membrane spans in proteins using an artificial neural network\n",
      "2009 IEEE Symposium on Computational Intelligence in Bioinformatics and Computational Biology 2009, 68-74. http://dx.doi.org/10.1109/CIBCB.2009.4925709 \n",
      "\n",
      "A. Bleckmann, J. Meiler,*\n",
      "Epothilones: Quantitative Structure Activity Relations Studied by Support Vector Machines and Artificial Neural Networks\n",
      "QSAR Comb. Sci. 2003, 22, 722-728. http://dx.doi.org/10.1002/qsar.200330837 \n",
      "\n",
      "A. Gulsevin, B. Han, J. Porta, H. Mchaourab, J. Meiler,* A. Kenworthy,\n",
      "Template-free prediction of a new monotopic membrane protein fold and oligomeric assembly by Alphafold2\n",
      "bioRxiv 2022. http://dx.doi.org/10.1101/2022.07.12.499809 \n",
      "\n",
      "R. Mueller, E. Dawson, J. Meiler,* A. Rodriguez, B. Chauder, B. Bates, A. Felts, J. Lamb, U. Menon, S. Jadhav, A. Kane, C. Jones, K. Gregory, C. Niswender, P. Conn, C. Olsen, D. Winder, K. Emmitte, C. Lindsley,\n",
      "Discovery of 2â(2âBenzoxazoyl amino)â4âArylâ5âCyanopyrimidine as Negative Allosteric Modulators (NAMs) of Metabotropic Glutamate Receptorâ5 (mGlu<sub>5</sub>): From an Artificial Neural Network Virtual Screen to an In Vivo Tool Compound\n",
      "ChemMedChem 2012, 7, 406-414. http://dx.doi.org/10.1002/cmdc.201100510 \n",
      "\n",
      "M. Butkiewicz, E. Lowe, J. Meiler,*\n",
      "Bcl&amp;#x2237;ChemInfo - Qualitative analysis of machine learning models for activation of HSD involved in Alzheimer's Disease\n",
      "2012 IEEE Symposium on Computational Intelligence in Bioinformatics and Computational Biology (CIBCB) 2012, 329-334. http://dx.doi.org/10.1109/CIBCB.2012.6217248 \n",
      "\n",
      "A. Gulsevin, J. Meiler,*\n",
      "Benchmarking Peptide Structure Prediction with AlphaFold2\n",
      "bioRxiv 2022. http://dx.doi.org/10.1101/2022.02.17.480937 \n",
      "\n",
      "A. Geanes, H. Cho, K. Nance, K. McGowan, P. Conn, C. Jones, J. Meiler,* C. Lindsley,\n",
      "Ligand-based virtual screen for the discovery of novel M5 inhibitor chemotypes\n",
      "Bioorganic &amp; Medicinal Chemistry Letters 2016, 26, 4487-4491. http://dx.doi.org/10.1016/j.bmcl.2016.07.071 \n",
      "\n",
      "V. Golkov, M. Skwark, A. Mirchev, G. Dikov, A. Geanes, J. Mendenhall, J. Meiler,* D. Cremers,\n",
      "3D Deep Learning for Biological Function Prediction from Physical Fields\n",
      "2020 International Conference on 3D Vision (3DV) 2021, 928-937. http://dx.doi.org/10.1109/3DV50981.2020.00103 \n",
      "\n",
      "J. Meiler,* R. Meusinger, M. Will,\n",
      "Fast Determination of <sup>13</sup>C NMR Chemical Shifts Using Artificial Neural Networks\n",
      "J. Chem. Inf. Comput. Sci. 2002, 40, 1169-1176. http://dx.doi.org/10.1021/ci000021c \n",
      "\n",
      "Y. Liu, Y. Wang, O. Vu, R. Moretti, B. Bodenheimer, J. Meiler,* T. Derr,\n",
      "Interpretable Chirality-Aware Graph Neural Network for Quantitative Structure Activity Relationship Modeling in Drug Discovery\n",
      "AAAI 2023, 37, 14356-14364. http://dx.doi.org/10.1609/aaai.v37i12.26679 \n",
      "\n",
      "D. del Alamo, D. Sala, H. Mchaourab, J. Meiler,*\n",
      "Sampling the conformational landscapes of transporters and receptors with AlphaFold2\n",
      "bioRxiv 2021. http://dx.doi.org/10.1101/2021.11.22.469536 \n",
      "\n",
      "J. Meiler,* W. Maier, M. Will, R. Meusinger,\n",
      "Using Neural Networks for 13C NMR Chemical Shift PredictionâComparison with Traditional Methods\n",
      "Journal of Magnetic Resonance 2002, 157, 242-252. http://dx.doi.org/10.1006/jmre.2002.2599 \n",
      "\n",
      "M. Ertelt, J. Meiler,* C. Schoeder,\n",
      "Combining Rosetta Sequence Design with Protein Language Model Predictions Using Evolutionary Scale Modeling (ESM) as Restraint\n",
      "ACS Synth. Biol. 2024, 13, 1085-1092. http://dx.doi.org/10.1021/acssynbio.3c00753 \n",
      "\n",
      "J. Mendenhall, J. Meiler,*\n",
      "Improving quantitative structureâactivity relationship models using Artificial Neural Networks trained with dropout\n",
      "J Comput Aided Mol Des 2016, 30, 177-189. http://dx.doi.org/10.1007/s10822-016-9895-2 \n",
      "\n",
      "J. Koehler, N. Woetzel, R. Staritzbichler, C. Sanders, J. Meiler,*\n",
      "A unified hydrophobicity scale for multispan membrane proteins\n",
      "Proteins 2008, 76, 13-29. http://dx.doi.org/10.1002/prot.22315 \n",
      "\n",
      "L. Somermeyer, A. Fleiss, A. Mishin, N. Bozhanova, A. Igolkina, J. Meiler,* M. Pujol, E. Putintseva, K. Sarkisyan, F. Kondrashov,\n",
      "Heterogeneity of the GFP fitness landscape and data-driven protein design\n",
      "bioRxiv 2021. http://dx.doi.org/10.1101/2021.12.08.471728 \n",
      "\n",
      "M. Ertelt, V. Mulligan, J. Maguire, S. Lyskov, R. Moretti, T. Schiffner, J. Meiler,* C. Schoeder,\n",
      "Combining machine learning with structure-based protein design to predict and engineer post-translational modifications of proteins\n",
      "PLoS Comput Biol 2024, 20, e1011939. http://dx.doi.org/10.1371/journal.pcbi.1011939 \n",
      "\n",
      "O. Vu, J. Mendenhall, D. Altarawy, J. Meiler,*\n",
      "BCL::Mol2Dâa robust atom environment descriptor for QSAR modeling and lead optimization\n",
      "J Comput Aided Mol Des 2019, 33, 477-486. http://dx.doi.org/10.1007/s10822-019-00199-8 \n",
      "\n",
      "J. Leman,, R. Mueller, M. Karakas, N. Woetzel, J. Meiler,*\n",
      "Simultaneous prediction of protein secondary structure and transmembrane spans\n",
      "Proteins 2013, 81, 1127-1140. http://dx.doi.org/10.1002/prot.24258 \n",
      "\n",
      "M. Fischer, J. Crowe, J. Meiler,*\n",
      "Computational epitope mapping of class I fusion proteins using low complexity supervised learning methods\n",
      "PLoS Comput Biol 2022, 18, e1010230. http://dx.doi.org/10.1371/journal.pcbi.1010230 \n",
      "\n",
      "E. Lowe, M. Butkiewicz, M. Spellings, A. Omlor, J. Meiler,*\n",
      "Comparative analysis of machine learning techniques for the prediction of logP\n",
      "2011 IEEE Symposium on Computational Intelligence in Bioinformatics and Computational Biology (CIBCB) 2011, 1-6. http://dx.doi.org/10.1109/CIBCB.2011.5948478 \n",
      "\n",
      "M. Butkiewicz, E. Lowe, R. Mueller, J. Mendenhall, P. Teixeira, C. Weaver, J. Meiler,*\n",
      "Benchmarking Ligand-Based Virtual High-Throughput Screening with the PubChem Database\n",
      "Molecules 2013, 18, 735-756. http://dx.doi.org/10.3390/molecules18010735 \n",
      "\n",
      "B. Brown, R. Stein, J. Meiler,* H. Mchaourab,\n",
      "Approximating conformational Boltzmann distributions with AlphaFold2 predictions\n",
      "bioRxiv 2023. http://dx.doi.org/10.1101/2023.08.06.552168 \n",
      "\n",
      "M. Kuhn, J. Meiler,* D. Baker,\n",
      "Strandâloopâstrand motifs: Prediction of hairpins and diverging turns in proteins\n",
      "Proteins 2003, 54, 282-288. http://dx.doi.org/10.1002/prot.10589 \n",
      "\n",
      "C. Kuntz, H. Woods, A. McKee, N. Zelt, J. Mendenhall, J. Meiler,* J. Schlebach,\n",
      "Towards Generalizable Predictions for the Effects of Mutations on G-Protein Coupled Receptor Expression\n",
      "bioRxiv 2021. http://dx.doi.org/10.1101/2021.12.28.474371 \n",
      "\n",
      "M. Ertelt, R. Moretti, J. Meiler,* C. Schoeder,\n",
      "Self-supervised machine learning methods for protein design improve sampling, but not the identification of high-fitness variants\n",
      "bioRxiv 2024. http://dx.doi.org/10.1101/2024.06.20.599843 \n",
      "\n",
      "E. Lowe, A. Ferrebee, A. Rodriguez, P. Conn, J. Meiler,*\n",
      "3D-QSAR CoMFA study of benzoxazepine derivatives as mGluR5 positive allosteric modulators\n",
      "Bioorganic &amp; Medicinal Chemistry Letters 2010, 20, 5922-5924. http://dx.doi.org/10.1016/j.bmcl.2010.07.061 \n",
      "\n",
      "J. Meiler,* M. KÃ¶ck,\n",
      "Novel methods of automated structure elucidation based on <sup>13</sup>C NMR spectroscopy\n",
      "Magnetic Reson in Chemistry 2004, 42, 1042-1045. http://dx.doi.org/10.1002/mrc.1424 \n",
      "\n",
      "C. Moth, J. Sheehan, A. Mamun, R. Sivley, A. Gulsevin, D. Rinker, J. Capra, J. Meiler,*\n",
      "VUStruct: a compute pipeline for high throughput and personalized structural biology\n",
      "bioRxiv 2024. http://dx.doi.org/10.1101/2024.08.06.606224 \n",
      "\n",
      "E. McDonald, K. Oliver, J. Schlebach, J. Meiler,* L. Plate,\n",
      "Benchmarking AlphaMissense Pathogenicity Predictions Against Cystic Fibrosis Variants\n",
      "bioRxiv 2023. http://dx.doi.org/10.1101/2023.10.05.561147 \n",
      "\n",
      "B. Brown, O. Vu, A. Geanes, S. Kothiwale, M. Butkiewicz, E. Lowe, R. Mueller, R. Pape, J. Mendenhall, J. Meiler,*\n",
      "Introduction to the BioChemical Library (BCL): An Application-Based Open-Source Toolkit for Integrated Cheminformatics and Machine Learning in Computer-Aided Drug Discovery\n",
      "Front. Pharmacol. 2022, 13. http://dx.doi.org/10.3389/fphar.2022.833099 \n",
      "\n",
      "J. Meiler,* R. Meusinger, M. Will,\n",
      "Neural Network Prediction of 13C NMR Chemical Shifts of Substituted Benzenes\n",
      "Monatshefte fuer Chemie 2006, 130, 1089-1095. http://dx.doi.org/10.1007/PL00010287 \n",
      "\n",
      "S. Phul, G. Kuenze, C. Vanoye, C. Sanders, A. George, J. Meiler,*\n",
      "Predicting the Functional Impact of KCNQ1 Variants with Artificial Neural Networks\n",
      "bioRxiv 2021. http://dx.doi.org/10.1101/2021.12.06.471347 \n",
      "\n",
      "B. Li, J. Mendenhall, J. Capra, J. Meiler,*\n",
      "A Multitask Deep-Learning Method for Predicting Membrane Associations and Secondary Structures of Proteins\n",
      "J. Proteome Res. 2021, 20, 4089-4100. http://dx.doi.org/10.1021/acs.jproteome.1c00410 \n",
      "\n",
      "S. DeLuca, J. Meiler,*\n",
      "A Novel Method for Guiding Protein-Ligand Docking with QSAR-Derived Pharmacophore Maps\n",
      "Biophysical Journal 2011, 100, 394a-395a. http://dx.doi.org/10.1016/j.bpj.2010.12.2344 \n",
      "\n",
      "D. del Alamo, L. DeSousa, R. Nair, S. Rahman, J. Meiler,* H. Mchaourab,\n",
      "Integrated AlphaFold2 and DEER investigation of the conformational dynamics of a pH-dependent APC antiporter\n",
      "Proc. Natl. Acad. Sci. U.S.A. 2022, 119. http://dx.doi.org/10.1073/pnas.2206129119 \n",
      "\n",
      "C. Jones, B. Brown, D. Schultz, J. Engers, V. Kramlinger, J. Meiler,* C. Lindsley,\n",
      "Computer-Aided Design and Biological Evaluation of Diazaspirocyclic D<sub>4</sub>R Antagonists\n",
      "ACS Chem. Neurosci. 2024, 15, 2396-2407. http://dx.doi.org/10.1021/acschemneuro.4c00086 \n",
      "\n",
      "L. Gonzalez Somermeyer, A. Fleiss, A. Mishin, N. Bozhanova, A. Igolkina, J. Meiler,* M. Alaball Pujol, E. Putintseva, K. Sarkisyan, F. Kondrashov,\n",
      "Heterogeneity of the GFP fitness landscape and data-driven protein design\n",
      "eLife 2022, 11. http://dx.doi.org/10.7554/eLife.75842 \n",
      "\n",
      "G. Sliwoski, S. Kothiwale, J. Meiler,* E. Lowe,\n",
      "Computational Methods in Drug Discovery\n",
      "Pharmacol Rev 2014, 66, 334-395. http://dx.doi.org/10.1124/pr.112.007336 \n",
      "\n",
      "E. McDonald, T. Jones, L. Plate, J. Meiler,* A. Gulsevin,\n",
      "Benchmarking AlphaFold2 on peptide structure prediction\n",
      "Structure 2022, 31, 111-119.e2. http://dx.doi.org/10.1016/j.str.2022.11.012 \n",
      "\n",
      "D. del Alamo, D. Sala, H. Mchaourab, J. Meiler,*\n",
      "Sampling alternative conformational states of transporters and receptors with AlphaFold2\n",
      "eLife 2022, 11. http://dx.doi.org/10.7554/eLife.75751 \n",
      "\n",
      "V. Golkov, A. Becker, D. Plop, D. Äuturilo, N. Davoudi, J. Mendenhall, R. Moretti, J. Meiler,* D. Cremers,\n",
      "Deep Learning for Virtual Screening: Five Reasons to Use ROC Cost Functions\n",
      "arXiv 2020. https://arxiv.org/abs/2007.07029 \n",
      "\n",
      "D. Sala, F. Engelberger, H. Mchaourab, J. Meiler,*\n",
      "Modeling conformational states of proteins with AlphaFold\n",
      "Current Opinion in Structural Biology 2023, 81, 102645. http://dx.doi.org/10.1016/j.sbi.2023.102645 \n",
      "\n",
      "E. McDonald, K. Oliver, J. Schlebach, J. Meiler,* L. Plate,\n",
      "Benchmarking AlphaMissense pathogenicity predictions against cystic fibrosis variants\n",
      "PLoS ONE 2024, 19, e0297560. http://dx.doi.org/10.1371/journal.pone.0297560 \n",
      "\n",
      "B. Brown, J. Mendenhall, A. Geanes, J. Meiler,*\n",
      "General Purpose Structure-Based Drug Discovery Neural Network Score Functions with Human-Interpretable Pharmacophore Maps\n",
      "J. Chem. Inf. Model. 2021, 61, 603-620. http://dx.doi.org/10.1021/acs.jcim.0c01001 \n",
      "\n",
      "M. Butkiewicz, R. Mueller, D. Selic, E. Dawson, J. Meiler,*\n",
      "Application of machine learning approaches on quantitative structure activity relationships\n",
      "2009 IEEE Symposium on Computational Intelligence in Bioinformatics and Computational Biology 2009, 255-262. http://dx.doi.org/10.1109/CIBCB.2009.4925736 \n",
      "\n",
      "B. Brown, R. Stein, J. Meiler,* H. Mchaourab,\n",
      "Approximating Projections of Conformational Boltzmann Distributions with AlphaFold2 Predictions: Opportunities and Limitations\n",
      "J. Chem. Theory Comput. 2024, 20, 1434-1447. http://dx.doi.org/10.1021/acs.jctc.3c01081 \n",
      "\n",
      "[['given', 'http://dx.doi.org/10.7554/eLife.92406.1', 'eLife Sciences Publications, Ltd'], ['given', 'https://arxiv.org/abs/2411.09820', 'arXiv']]\n"
     ]
    }
   ],
   "source": [
    "errors = []\n",
    "\n",
    "for doi in list(ai_dois):\n",
    "    ref = all_refs[doi]\n",
    "    try:\n",
    "        #if 'created' in ref:\n",
    "        #    if ref['created']['date-parts'][0][0] < 2022:\n",
    "        #        continue\n",
    "        #elif 'issued' in ref:\n",
    "        #    if ref['issued']['date-parts'][0][0] < 2022:\n",
    "        #        continue\n",
    "        citation, preprint = formatSPP(ref)\n",
    "        #if not preprint:\n",
    "        #    print(citation, '\\n')\n",
    "        print(citation, '\\n')\n",
    "    except KeyError as e:\n",
    "        errors.append([e.args[0], ref['URL'], ref['publisher']])\n",
    "print(errors)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " @article{Kothiwale_2017, title={Quantitative StructureÃ¢ÂÂActivity Relationship Modeling of Kinase Selectivity Profiles}, volume={22}, ISSN={1420-3049}, url={http://dx.doi.org/10.3390/molecules22091576}, DOI={10.3390/molecules22091576}, number={9}, journal={Molecules}, publisher={MDPI AG}, author={Kothiwale, Sandeepkumar and Borza, Corina and Pozzi, Ambra and Meiler, Jens}, year={2017}, month=sep, pages={1576} }\n",
      "\n",
      " @article{Liu_2023, title={Advancements in Ligand-Based Virtual Screening through the Synergistic Integration of Graph Neural Networks and Expert-Crafted Descriptors}, url={http://dx.doi.org/10.1101/2023.04.17.537185}, DOI={10.1101/2023.04.17.537185}, publisher={Cold Spring Harbor Laboratory}, author={Liu, Yunchao (Lance) and Moretti, Rocco and Wang, Yu and Dong, Ha and Yan, Bailu and Bodenheimer, Bobby and Derr, Tyler and Meiler, Jens}, year={2023}, month=apr }\n",
      "\n",
      " @article{DeCorte_2024, title={Interpretable deep-learning pKa prediction for small molecule drugs via atomic sensitivity analysis}, url={http://dx.doi.org/10.26434/chemrxiv-2024-hr692}, DOI={10.26434/chemrxiv-2024-hr692}, publisher={American Chemical Society (ACS)}, author={DeCorte, Joseph and Brown, Benjamin and Meiler, Jens}, year={2024}, month=jun }\n",
      "\n",
      " @article{Phul_2022, title={Predicting the functional impact of KCNQ1 variants with artificial neural networks}, volume={18}, ISSN={1553-7358}, url={http://dx.doi.org/10.1371/journal.pcbi.1010038}, DOI={10.1371/journal.pcbi.1010038}, number={4}, journal={PLOS Computational Biology}, publisher={Public Library of Science (PLoS)}, author={Phul, Saksham and Kuenze, Georg and Vanoye, Carlos G. and Sanders, Charles R. and George, Alfred L. and Meiler, Jens}, editor={Slusky, Joanna}, year={2022}, month=apr, pages={e1010038} }\n",
      "\n",
      " @article{Sala_2023, title={Biasing AlphaFold2 to predict GPCRs and kinases with user-defined functional or structural properties}, volume={10}, ISSN={2296-889X}, url={http://dx.doi.org/10.3389/fmolb.2023.1121962}, DOI={10.3389/fmolb.2023.1121962}, journal={Frontiers in Molecular Biosciences}, publisher={Frontiers Media SA}, author={Sala, Davide and Hildebrand, Peter W. and Meiler, Jens}, year={2023}, month=feb }\n",
      "\n",
      " @article{Gulsevin_2023, title={Template-free prediction of a new monotopic membrane protein fold and assembly by AlphaFold2}, volume={122}, ISSN={0006-3495}, url={http://dx.doi.org/10.1016/j.bpj.2022.11.011}, DOI={10.1016/j.bpj.2022.11.011}, number={11}, journal={Biophysical Journal}, publisher={Elsevier BV}, author={Gulsevin, Alican and Han, Bing and Porta, Jason C. and Mchaourab, Hassane S. and Meiler, Jens and Kenworthy, Anne K.}, year={2023}, month=jun, pages={2041Ã¢ÂÂ2052} }\n",
      "\n",
      " @article{Meiler_2003, title={Coupled prediction of protein secondary and tertiary structure}, volume={100}, ISSN={1091-6490}, url={http://dx.doi.org/10.1073/pnas.1831973100}, DOI={10.1073/pnas.1831973100}, number={21}, journal={Proceedings of the National Academy of Sciences}, publisher={Proceedings of the National Academy of Sciences}, author={Meiler, Jens and Baker, David}, year={2003}, month=oct, pages={12105Ã¢ÂÂ12110} }\n",
      "\n",
      " @article{Sliwoski_2015, title={Autocorrelation descriptor improvements for QSAR: 2DA_Sign and 3DA_Sign}, volume={30}, ISSN={1573-4951}, url={http://dx.doi.org/10.1007/s10822-015-9893-9}, DOI={10.1007/s10822-015-9893-9}, number={3}, journal={Journal of Computer-Aided Molecular Design}, publisher={Springer Science and Business Media LLC}, author={Sliwoski, Gregory and Mendenhall, Jeffrey and Meiler, Jens}, year={2015}, month=dec, pages={209Ã¢ÂÂ217} }\n",
      "\n",
      " @article{Meiler_2001, title={Automated Structure Elucidation of Organic Molecules from 13C NMR Spectra Using Genetic Algorithms and Neural Networks}, volume={41}, ISSN={1520-5142}, url={http://dx.doi.org/10.1021/ci0102970}, DOI={10.1021/ci0102970}, number={6}, journal={Journal of Chemical Information and Computer Sciences}, publisher={American Chemical Society (ACS)}, author={Meiler, Jens and Will, Martin}, year={2001}, month=oct, pages={1535Ã¢ÂÂ1546} }\n",
      "\n",
      " @article{Kuang_2024, title={SuperWater: Predicting Water Molecule Positions on Protein Structures by Generative AI}, url={http://dx.doi.org/10.1101/2024.11.18.624208}, DOI={10.1101/2024.11.18.624208}, publisher={Cold Spring Harbor Laboratory}, author={Kuang, Xiaohan and su, zhaoqian and Liu, Yunchao and Lin, Xiaobo and Smith, Jesse Spencer and Derr, Tyler and Wu, Yinghao and Meiler, Jens}, year={2024}, month=nov }\n",
      "\n",
      " @article{Sliwoski_2012, title={BCL::EMAS Ã¢ÂÂ Enantioselective Molecular Asymmetry Descriptor for 3D-QSAR}, volume={17}, ISSN={1420-3049}, url={http://dx.doi.org/10.3390/molecules17089971}, DOI={10.3390/molecules17089971}, number={8}, journal={Molecules}, publisher={MDPI AG}, author={Sliwoski, Gregory and Lowe, Edward W. and Butkiewicz, Mariusz and Meiler, Jens}, year={2012}, month=aug, pages={9971Ã¢ÂÂ9989} }\n",
      "\n",
      " @article{Lowe_2012, title={GPU-accelerated machine learning techniques enable QSAR modeling of large HTS data}, url={http://dx.doi.org/10.1109/CIBCB.2012.6217246}, DOI={10.1109/cibcb.2012.6217246}, journal={2012 IEEE Symposium on Computational Intelligence in Bioinformatics and Computational Biology (CIBCB)}, publisher={IEEE}, author={Lowe, E. W. and Butkiewicz, M. and Woetzel, N. and Meiler, J.}, year={2012}, month=may, pages={314Ã¢ÂÂ320} }\n",
      "\n",
      " @article{Blee_2022, title={An Active Learning Framework Improves Tumor Variant Interpretation}, volume={82}, ISSN={1538-7445}, url={http://dx.doi.org/10.1158/0008-5472.CAN-21-3798}, DOI={10.1158/0008-5472.can-21-3798}, number={15}, journal={Cancer Research}, publisher={American Association for Cancer Research (AACR)}, author={Blee, Alexandra M. and Li, Bian and Pecen, Turner and Meiler, Jens and Nagel, Zachary D. and Capra, John A. and Chazin, Walter J.}, year={2022}, month=jun, pages={2704Ã¢ÂÂ2715} }\n",
      "\n",
      " @article{Jiang_2024, title={Machine learning application to predict binding affinity between peptide containing non-canonical amino acids and HLA0201}, url={http://dx.doi.org/10.1101/2024.11.19.624425}, DOI={10.1101/2024.11.19.624425}, publisher={Cold Spring Harbor Laboratory}, author={Jiang, Shan and Su, Zhaoqian and Bloodworth, Nathaniel and Liu, Yunchao and Martina, Cristina and Harrison, David G. and Meiler, Jens}, year={2024}, month=nov }\n",
      "\n",
      " @article{Koehler_2009, title={Improved prediction of trans-membrane spans in proteins using an artificial neural network}, url={http://dx.doi.org/10.1109/CIBCB.2009.4925709}, DOI={10.1109/cibcb.2009.4925709}, journal={2009 IEEE Symposium on Computational Intelligence in Bioinformatics and Computational Biology}, publisher={IEEE}, author={Koehler, Julia and Mueller, Ralf and Meiler, Jens}, year={2009}, month=mar, pages={68Ã¢ÂÂ74} }\n",
      "\n",
      " @article{Bleckmann_2003, title={Epothilones: Quantitative Structure Activity Relations Studied by Support Vector Machines and Artificial Neural Networks}, volume={22}, ISSN={1611-0218}, url={http://dx.doi.org/10.1002/qsar.200330837}, DOI={10.1002/qsar.200330837}, number={7}, journal={QSAR &amp; Combinatorial Science}, publisher={Wiley}, author={Bleckmann, Annalen and Meiler, Jens}, year={2003}, month=oct, pages={722Ã¢ÂÂ728} }\n",
      "\n",
      " @article{Gulsevin_2022, title={Template-free prediction of a new monotopic membrane protein fold and oligomeric assembly by Alphafold2}, url={http://dx.doi.org/10.1101/2022.07.12.499809}, DOI={10.1101/2022.07.12.499809}, publisher={Cold Spring Harbor Laboratory}, author={Gulsevin, Alican and Han, Bing and Porta, Jason C. and Mchaourab, Hassane S. and Meiler, Jens and Kenworthy, Anne K.}, year={2022}, month=jul }\n",
      "\n",
      " @article{Mueller_2012, title={Discovery of 2Ã¢ÂÂ(2Ã¢ÂÂBenzoxazoyl amino)Ã¢ÂÂ4Ã¢ÂÂArylÃ¢ÂÂ5Ã¢ÂÂCyanopyrimidine as Negative Allosteric Modulators (NAMs) of Metabotropic Glutamate ReceptorÃ¢ÂÂ5 (mGlu5): From an Artificial Neural Network Virtual Screen to an In Vivo Tool Compound}, volume={7}, ISSN={1860-7187}, url={http://dx.doi.org/10.1002/cmdc.201100510}, DOI={10.1002/cmdc.201100510}, number={3}, journal={ChemMedChem}, publisher={Wiley}, author={Mueller, Ralf and Dawson, Eric S. and Meiler, Jens and Rodriguez, Alice L. and Chauder, Brian A. and Bates, Brittney S. and Felts, Andrew S. and Lamb, Jeffrey P. and Menon, Usha N. and Jadhav, Sataywan B. and Kane, Alexander S. and Jones, Carrie K. and Gregory, Karen J. and Niswender, Colleen M. and Conn, P. Jeffrey and Olsen, Christopher M. and Winder, Danny G. and Emmitte, Kyle A. and Lindsley, Craig W.}, year={2012}, month=jan, pages={406Ã¢ÂÂ414} }\n",
      "\n",
      " @article{Butkiewicz_2012, title={Bcl&amp;#x2237;ChemInfo - Qualitative analysis of machine learning models for activation of HSD involved in AlzheimerÃ¢ÂÂs Disease}, url={http://dx.doi.org/10.1109/CIBCB.2012.6217248}, DOI={10.1109/cibcb.2012.6217248}, journal={2012 IEEE Symposium on Computational Intelligence in Bioinformatics and Computational Biology (CIBCB)}, publisher={IEEE}, author={Butkiewicz, Mariusz and Lowe, Edward W. and Meiler, Jens}, year={2012}, month=may, pages={329Ã¢ÂÂ334} }\n",
      "\n",
      " @article{Chamness_2023, title={Divergent Pairwise Epistasis in the Context of Unstable Membrane Protein Variants}, url={http://dx.doi.org/10.7554/eLife.92406.1}, DOI={10.7554/elife.92406.1}, publisher={eLife Sciences Publications, Ltd}, author={Chamness, Laura M. and Kuntz, Charles P. and McKee, Andrew G. and Penn, Wesley D. and Hemmerich, Christopher M. and Rusch, Douglas B. and Woods, Hope and Dyotima and Meiler, Jens and Schlebach, Jonathan P.}, year={2023}, month=dec }\n",
      "\n",
      " @article{Gulsevin_2022, title={Benchmarking Peptide Structure Prediction with AlphaFold2}, url={http://dx.doi.org/10.1101/2022.02.17.480937}, DOI={10.1101/2022.02.17.480937}, publisher={Cold Spring Harbor Laboratory}, author={Gulsevin, Alican and Meiler, Jens}, year={2022}, month=feb }\n",
      "\n",
      " @article{Geanes_2016, title={Ligand-based virtual screen for the discovery of novel M5 inhibitor chemotypes}, volume={26}, ISSN={0960-894X}, url={http://dx.doi.org/10.1016/j.bmcl.2016.07.071}, DOI={10.1016/j.bmcl.2016.07.071}, number={18}, journal={Bioorganic &amp; Medicinal Chemistry Letters}, publisher={Elsevier BV}, author={Geanes, Alexander R. and Cho, Hykeyung P. and Nance, Kellie D. and McGowan, Kevin M. and Conn, P. Jeffrey and Jones, Carrie K. and Meiler, Jens and Lindsley, Craig W.}, year={2016}, month=sep, pages={4487Ã¢ÂÂ4491} }\n",
      "\n",
      " @article{Golkov_2020, title={3D Deep Learning for Biological Function Prediction from Physical Fields}, url={http://dx.doi.org/10.1109/3DV50981.2020.00103}, DOI={10.1109/3dv50981.2020.00103}, journal={2020 International Conference on 3D Vision (3DV)}, publisher={IEEE}, author={Golkov, Vladimir and Skwark, Marcin J. and Mirchev, Atanas and Dikov, Georgi and Geanes, Alexander R. and Mendenhall, Jeffrey and Meiler, Jens and Cremers, Daniel}, year={2020}, month=nov, pages={928Ã¢ÂÂ937} }\n",
      "\n",
      " @article{Meiler_2000, title={Fast Determination of 13C NMR Chemical Shifts Using Artificial Neural Networks}, volume={40}, ISSN={0095-2338}, url={http://dx.doi.org/10.1021/ci000021c}, DOI={10.1021/ci000021c}, number={5}, journal={Journal of Chemical Information and Computer Sciences}, publisher={American Chemical Society (ACS)}, author={Meiler, J. and Meusinger, R. and Will, M.}, year={2000}, month=aug, pages={1169Ã¢ÂÂ1176} }\n",
      "\n",
      " @article{Liu_2023, title={Interpretable Chirality-Aware Graph Neural Network for Quantitative Structure Activity Relationship Modeling in Drug Discovery}, volume={37}, ISSN={2159-5399}, url={http://dx.doi.org/10.1609/aaai.v37i12.26679}, DOI={10.1609/aaai.v37i12.26679}, number={12}, journal={Proceedings of the AAAI Conference on Artificial Intelligence}, publisher={Association for the Advancement of Artificial Intelligence (AAAI)}, author={Liu, Yunchao (Lance) and Wang, Yu and Vu, Oanh and Moretti, Rocco and Bodenheimer, Bobby and Meiler, Jens and Derr, Tyler}, year={2023}, month=jun, pages={14356Ã¢ÂÂ14364} }\n",
      "\n",
      " @article{del_Alamo_2021, title={Sampling the conformational landscapes of transporters and receptors with AlphaFold2}, url={http://dx.doi.org/10.1101/2021.11.22.469536}, DOI={10.1101/2021.11.22.469536}, publisher={Cold Spring Harbor Laboratory}, author={del Alamo, Diego and Sala, Davide and Mchaourab, Hassane S. and Meiler, Jens}, year={2021}, month=nov }\n",
      "\n",
      " @article{Meiler_2002, title={Using Neural Networks for 13C NMR Chemical Shift PredictionÃ¢ÂÂComparison with Traditional Methods}, volume={157}, ISSN={1090-7807}, url={http://dx.doi.org/10.1006/jmre.2002.2599}, DOI={10.1006/jmre.2002.2599}, number={2}, journal={Journal of Magnetic Resonance}, publisher={Elsevier BV}, author={Meiler, Jens and Maier, Walter and Will, Martin and Meusinger, Reinhard}, year={2002}, month=aug, pages={242Ã¢ÂÂ252} }\n",
      "\n",
      " @article{Ertelt_2024, title={Combining Rosetta Sequence Design with Protein Language Model Predictions Using Evolutionary Scale Modeling (ESM) as Restraint}, volume={13}, ISSN={2161-5063}, url={http://dx.doi.org/10.1021/acssynbio.3c00753}, DOI={10.1021/acssynbio.3c00753}, number={4}, journal={ACS Synthetic Biology}, publisher={American Chemical Society (ACS)}, author={Ertelt, Moritz and Meiler, Jens and Schoeder, Clara T.}, year={2024}, month=apr, pages={1085Ã¢ÂÂ1092} }\n",
      "\n",
      " @article{Mendenhall_2016, title={Improving quantitative structureÃ¢ÂÂactivity relationship models using Artificial Neural Networks trained with dropout}, volume={30}, ISSN={1573-4951}, url={http://dx.doi.org/10.1007/s10822-016-9895-2}, DOI={10.1007/s10822-016-9895-2}, number={2}, journal={Journal of Computer-Aided Molecular Design}, publisher={Springer Science and Business Media LLC}, author={Mendenhall, Jeffrey and Meiler, Jens}, year={2016}, month=feb, pages={177Ã¢ÂÂ189} }\n",
      "\n",
      " @article{Koehler_2008, title={A unified hydrophobicity scale for multispan membrane proteins}, volume={76}, ISSN={1097-0134}, url={http://dx.doi.org/10.1002/prot.22315}, DOI={10.1002/prot.22315}, number={1}, journal={Proteins: Structure, Function, and Bioinformatics}, publisher={Wiley}, author={Koehler, Julia and Woetzel, Nils and Staritzbichler, RenÃÂ© and Sanders, Charles R. and Meiler, Jens}, year={2008}, month=dec, pages={13Ã¢ÂÂ29} }\n",
      "\n",
      " @article{Somermeyer_2021, title={Heterogeneity of the GFP fitness landscape and data-driven protein design}, url={http://dx.doi.org/10.1101/2021.12.08.471728}, DOI={10.1101/2021.12.08.471728}, publisher={Cold Spring Harbor Laboratory}, author={Somermeyer, Louisa Gonzalez and Fleiss, Aubin and Mishin, Alexander S. and Bozhanova, Nina G. and Igolkina, Anna A. and Meiler, Jens and Pujol, Maria-Elisenda Alaball and Putintseva, Ekaterina V. and Sarkisyan, Karen S. and Kondrashov, Fyodor A.}, year={2021}, month=dec }\n",
      "\n",
      " @article{Ertelt_2024, title={Combining machine learning with structure-based protein design to predict and engineer post-translational modifications of proteins}, volume={20}, ISSN={1553-7358}, url={http://dx.doi.org/10.1371/journal.pcbi.1011939}, DOI={10.1371/journal.pcbi.1011939}, number={3}, journal={PLOS Computational Biology}, publisher={Public Library of Science (PLoS)}, author={Ertelt, Moritz and Mulligan, Vikram Khipple and Maguire, Jack B. and Lyskov, Sergey and Moretti, Rocco and Schiffner, Torben and Meiler, Jens and Schoeder, Clara T.}, editor={Slusky, Joanna}, year={2024}, month=mar, pages={e1011939} }\n",
      "\n",
      " @article{Vu_2019, title={BCL::Mol2DÃ¢ÂÂa robust atom environment descriptor for QSAR modeling and lead optimization}, volume={33}, ISSN={1573-4951}, url={http://dx.doi.org/10.1007/s10822-019-00199-8}, DOI={10.1007/s10822-019-00199-8}, number={5}, journal={Journal of Computer-Aided Molecular Design}, publisher={Springer Science and Business Media LLC}, author={Vu, Oanh and Mendenhall, Jeffrey and Altarawy, Doaa and Meiler, Jens}, year={2019}, month=apr, pages={477Ã¢ÂÂ486} }\n",
      "\n",
      " @article{Leman__2013,_Mueller_Karakas_Woetzel_Meiler_2013, title={Simultaneous prediction of protein secondary structure and transmembrane spans}, volume={81}, ISSN={1097-0134}, url={http://dx.doi.org/10.1002/prot.24258}, DOI={10.1002/prot.24258}, number={7}, journal={Proteins: Structure, Function, and Bioinformatics}, publisher={Wiley}, author={Leman, Julia Koehler and Mueller, Ralf and Karakas, Mert and Woetzel, Nils and Meiler, Jens}, year={2013}, month=apr, pages={1127Ã¢ÂÂ1140} }\n",
      "\n",
      " @article{Fischer_2022, title={Computational epitope mapping of class I fusion proteins using low complexity supervised learning methods}, volume={18}, ISSN={1553-7358}, url={http://dx.doi.org/10.1371/journal.pcbi.1010230}, DOI={10.1371/journal.pcbi.1010230}, number={12}, journal={PLOS Computational Biology}, publisher={Public Library of Science (PLoS)}, author={Fischer, Marion F. S. and Crowe, James E. and Meiler, Jens}, editor={Schneidman, Dina}, year={2022}, month=dec, pages={e1010230} }\n",
      "\n",
      " @article{Lowe_2011, title={Comparative analysis of machine learning techniques for the prediction of logP}, url={http://dx.doi.org/10.1109/CIBCB.2011.5948478}, DOI={10.1109/cibcb.2011.5948478}, journal={2011 IEEE Symposium on Computational Intelligence in Bioinformatics and Computational Biology (CIBCB)}, publisher={IEEE}, author={Lowe, Edward W. and Butkiewicz, Mariusz and Spellings, Matthew and Omlor, Albert and Meiler, Jens}, year={2011}, month=apr, pages={1Ã¢ÂÂ6} }\n",
      "\n",
      " @article{Butkiewicz_2013, title={Benchmarking Ligand-Based Virtual High-Throughput Screening with the PubChem Database}, volume={18}, ISSN={1420-3049}, url={http://dx.doi.org/10.3390/molecules18010735}, DOI={10.3390/molecules18010735}, number={1}, journal={Molecules}, publisher={MDPI AG}, author={Butkiewicz, Mariusz and Lowe, Edward and Mueller, Ralf and Mendenhall, Jeffrey and Teixeira, Pedro and Weaver, C. and Meiler, Jens}, year={2013}, month=jan, pages={735Ã¢ÂÂ756} }\n",
      "\n",
      " @article{Brown_2023, title={Approximating conformational Boltzmann distributions with AlphaFold2 predictions}, url={http://dx.doi.org/10.1101/2023.08.06.552168}, DOI={10.1101/2023.08.06.552168}, publisher={Cold Spring Harbor Laboratory}, author={Brown, Benjamin P. and Stein, Richard A. and Meiler, Jens and Mchaourab, Hassane}, year={2023}, month=aug }\n",
      "\n",
      " @article{Kuhn_2003, title={StrandÃ¢ÂÂloopÃ¢ÂÂstrand motifs: Prediction of hairpins and diverging turns in proteins}, volume={54}, ISSN={1097-0134}, url={http://dx.doi.org/10.1002/prot.10589}, DOI={10.1002/prot.10589}, number={2}, journal={Proteins: Structure, Function, and Bioinformatics}, publisher={Wiley}, author={Kuhn, Michael and Meiler, Jens and Baker, David}, year={2003}, month=dec, pages={282Ã¢ÂÂ288} }\n",
      "\n",
      " @article{Kuntz_2021, title={Towards Generalizable Predictions for the Effects of Mutations on G-Protein Coupled Receptor Expression}, url={http://dx.doi.org/10.1101/2021.12.28.474371}, DOI={10.1101/2021.12.28.474371}, publisher={Cold Spring Harbor Laboratory}, author={Kuntz, Charles P. and Woods, Hope and McKee, Andrew G. and Zelt, Nathan B. and Mendenhall, Jeffrey L. and Meiler, Jens and Schlebach, Jonathan P.}, year={2021}, month=dec }\n",
      "\n",
      " @article{Ertelt_2024, title={Self-supervised machine learning methods for protein design improve sampling, but not the identification of high-fitness variants}, url={http://dx.doi.org/10.1101/2024.06.20.599843}, DOI={10.1101/2024.06.20.599843}, publisher={Cold Spring Harbor Laboratory}, author={Ertelt, Moritz and Moretti, Rocco and Meiler, Jens and Schoeder, Clara T.}, year={2024}, month=jun }\n",
      "\n",
      " @article{Lowe_2010, title={3D-QSAR CoMFA study of benzoxazepine derivatives as mGluR5 positive allosteric modulators}, volume={20}, ISSN={0960-894X}, url={http://dx.doi.org/10.1016/j.bmcl.2010.07.061}, DOI={10.1016/j.bmcl.2010.07.061}, number={19}, journal={Bioorganic &amp; Medicinal Chemistry Letters}, publisher={Elsevier BV}, author={Lowe, Edward W. and Ferrebee, Alysia and Rodriguez, Alice L. and Conn, P. Jeffrey and Meiler, Jens}, year={2010}, month=oct, pages={5922Ã¢ÂÂ5924} }\n",
      "\n",
      " @article{Meiler_2004, title={Novel methods of automated structure elucidation based on 13C NMR spectroscopy}, volume={42}, ISSN={1097-458X}, url={http://dx.doi.org/10.1002/mrc.1424}, DOI={10.1002/mrc.1424}, number={12}, journal={Magnetic Resonance in Chemistry}, publisher={Wiley}, author={Meiler, Jens and KÃÂ¶ck, Matthias}, year={2004}, month=oct, pages={1042Ã¢ÂÂ1045} }\n",
      "\n",
      " @article{Moth_2024, title={VUStruct: a compute pipeline for high throughput and personalized structural biology}, url={http://dx.doi.org/10.1101/2024.08.06.606224}, DOI={10.1101/2024.08.06.606224}, publisher={Cold Spring Harbor Laboratory}, author={Moth, Christopher W. and Sheehan, Jonathan H. and Mamun, Abdullah Al and Sivley, R. Michael and Gulsevin, Alican and Rinker, David and Capra, John A. and Meiler, Jens}, year={2024}, month=aug }\n",
      "\n",
      " @article{McDonald_2023, title={Benchmarking AlphaMissense Pathogenicity Predictions Against Cystic Fibrosis Variants}, url={http://dx.doi.org/10.1101/2023.10.05.561147}, DOI={10.1101/2023.10.05.561147}, publisher={Cold Spring Harbor Laboratory}, author={McDonald, Eli Fritz and Oliver, Kathryn E. and Schlebach, Jonathan P. and Meiler, Jens and Plate, Lars}, year={2023}, month=oct }\n",
      "\n",
      " @article{Brown_2022, title={Introduction to the BioChemical Library (BCL): An Application-Based Open-Source Toolkit for Integrated Cheminformatics and Machine Learning in Computer-Aided Drug Discovery}, volume={13}, ISSN={1663-9812}, url={http://dx.doi.org/10.3389/fphar.2022.833099}, DOI={10.3389/fphar.2022.833099}, journal={Frontiers in Pharmacology}, publisher={Frontiers Media SA}, author={Brown, Benjamin P. and Vu, Oanh and Geanes, Alexander R. and Kothiwale, Sandeepkumar and Butkiewicz, Mariusz and Lowe, Edward W. and Mueller, Ralf and Pape, Richard and Mendenhall, Jeffrey and Meiler, Jens}, year={2022}, month=feb }\n",
      "\n",
      " @article{Meiler_1999, title={Neural Network Prediction of 13C NMR Chemical Shifts of Substituted Benzenes}, volume={130}, ISSN={1434-4475}, url={http://dx.doi.org/10.1007/PL00010287}, DOI={10.1007/pl00010287}, number={9}, journal={Monatshefte fÃÂ¼r Chemie / Chemical Monthly}, publisher={Springer Science and Business Media LLC}, author={Meiler, Jens and Meusinger, Reinhard and Will, Martin}, year={1999}, month=sep, pages={1089Ã¢ÂÂ1095} }\n",
      "\n",
      " @article{Phul_2021, title={Predicting the Functional Impact of KCNQ1 Variants with Artificial Neural Networks}, url={http://dx.doi.org/10.1101/2021.12.06.471347}, DOI={10.1101/2021.12.06.471347}, publisher={Cold Spring Harbor Laboratory}, author={Phul, Saksham and Kuenze, Georg and Vanoye, Carlos G. and Sanders, Charles R. and George, Alfred L. and Meiler, Jens}, year={2021}, month=dec }\n",
      "\n",
      " @article{Li_2021, title={A Multitask Deep-Learning Method for Predicting Membrane Associations and Secondary Structures of Proteins}, volume={20}, ISSN={1535-3907}, url={http://dx.doi.org/10.1021/acs.jproteome.1c00410}, DOI={10.1021/acs.jproteome.1c00410}, number={8}, journal={Journal of Proteome Research}, publisher={American Chemical Society (ACS)}, author={Li, Bian and Mendenhall, Jeffrey and Capra, John A. and Meiler, Jens}, year={2021}, month=jul, pages={4089Ã¢ÂÂ4100} }\n",
      "\n",
      " @article{DeLuca_2011, title={A Novel Method for Guiding Protein-Ligand Docking with QSAR-Derived Pharmacophore Maps}, volume={100}, ISSN={0006-3495}, url={http://dx.doi.org/10.1016/j.bpj.2010.12.2344}, DOI={10.1016/j.bpj.2010.12.2344}, number={3}, journal={Biophysical Journal}, publisher={Elsevier BV}, author={DeLuca, Sam and Meiler, Jens}, year={2011}, month=feb, pages={394aÃ¢ÂÂ395a} }\n",
      "\n",
      " @article{del_Alamo_2022, title={Integrated AlphaFold2 and DEER investigation of the conformational dynamics of a pH-dependent APC antiporter}, volume={119}, ISSN={1091-6490}, url={http://dx.doi.org/10.1073/pnas.2206129119}, DOI={10.1073/pnas.2206129119}, number={34}, journal={Proceedings of the National Academy of Sciences}, publisher={Proceedings of the National Academy of Sciences}, author={del Alamo, Diego and DeSousa, Lillian and Nair, Rahul M. and Rahman, Suhaila and Meiler, Jens and Mchaourab, Hassane S.}, year={2022}, month=aug }\n",
      "\n",
      " @article{yunchao_liu_dong_wang_moretti_wang_su_gu_bodenheimer_weaver_et al._2024, title={WelQrate: Defining the Gold Standard in Small Molecule Drug Discovery Benchmarking}, url={https://arxiv.org/abs/2411.09820}, DOI={10.48550/ARXIV.2411.09820}, abstractNote={While deep learning has revolutionized computer-aided drug discovery, the AI community has predominantly focused on model innovation and placed less emphasis on establishing best benchmarking practices. We posit that without a sound model evaluation framework, the AI community's efforts cannot reach their full potential, thereby slowing the progress and transfer of innovation into real-world drug discovery. Thus, in this paper, we seek to establish a new gold standard for small molecule drug discovery benchmarking, WelQrate. Specifically, our contributions are threefold: WelQrate Dataset Collection - we introduce a meticulously curated collection of 9 datasets spanning 5 therapeutic target classes. Our hierarchical curation pipelines, designed by drug discovery experts, go beyond the primary high-throughput screen by leveraging additional confirmatory and counter screens along with rigorous domain-driven preprocessing, such as Pan-Assay Interference Compounds (PAINS) filtering, to ensure the high-quality data in the datasets; WelQrate Evaluation Framework - we propose a standardized model evaluation framework considering high-quality datasets, featurization, 3D conformation generation, evaluation metrics, and data splits, which provides a reliable benchmarking for drug discovery experts conducting real-world virtual screening; Benchmarking - we evaluate model performance through various research questions using the WelQrate dataset collection, exploring the effects of different models, dataset quality, featurization methods, and data splitting strategies on the results. In summary, we recommend adopting our proposed WelQrate as the gold standard in small molecule drug discovery benchmarking. The WelQrate dataset collection, along with the curation codes, and experimental scripts are all publicly available at WelQrate.org.}, publisher={arXiv}, author={Yunchao and Liu and Dong, Ha and Wang, Xin and Moretti, Rocco and Wang, Yu and Su, Zhaoqian and Gu, Jiawei and Bodenheimer, Bobby and Weaver, Charles David and et al.}, year={2024} }\n",
      " @article{Jones_2024, title={Computer-Aided Design and Biological Evaluation of Diazaspirocyclic D4R Antagonists}, volume={15}, ISSN={1948-7193}, url={http://dx.doi.org/10.1021/acschemneuro.4c00086}, DOI={10.1021/acschemneuro.4c00086}, number={12}, journal={ACS Chemical Neuroscience}, publisher={American Chemical Society (ACS)}, author={Jones, Caleb A. H. and Brown, Benjamin P. and Schultz, Daniel C. and Engers, Julie and Kramlinger, Valerie M. and Meiler, Jens and Lindsley, Craig W.}, year={2024}, month=jun, pages={2396Ã¢ÂÂ2407} }\n",
      "\n",
      " @article{Gonzalez_Somermeyer_2022, title={Heterogeneity of the GFP fitness landscape and data-driven protein design}, volume={11}, ISSN={2050-084X}, url={http://dx.doi.org/10.7554/eLife.75842}, DOI={10.7554/elife.75842}, journal={eLife}, publisher={eLife Sciences Publications, Ltd}, author={Gonzalez Somermeyer, Louisa and Fleiss, Aubin and Mishin, Alexander S and Bozhanova, Nina G and Igolkina, Anna A and Meiler, Jens and Alaball Pujol, Maria-Elisenda and Putintseva, Ekaterina V and Sarkisyan, Karen S and Kondrashov, Fyodor A}, year={2022}, month=may }\n",
      "\n",
      " @article{Sliwoski_2013, title={Computational Methods in Drug Discovery}, volume={66}, ISSN={1521-0081}, url={http://dx.doi.org/10.1124/pr.112.007336}, DOI={10.1124/pr.112.007336}, number={1}, journal={Pharmacological Reviews}, publisher={American Society for Pharmacology & Experimental Therapeutics (ASPET)}, author={Sliwoski, Gregory and Kothiwale, Sandeepkumar and Meiler, Jens and Lowe, Edward W.}, editor={Barker, Eric L.}, year={2013}, month=dec, pages={334Ã¢ÂÂ395} }\n",
      "\n",
      " @article{McDonald_2023, title={Benchmarking AlphaFold2 on peptide structure prediction}, volume={31}, ISSN={0969-2126}, url={http://dx.doi.org/10.1016/j.str.2022.11.012}, DOI={10.1016/j.str.2022.11.012}, number={1}, journal={Structure}, publisher={Elsevier BV}, author={McDonald, Eli Fritz and Jones, Taylor and Plate, Lars and Meiler, Jens and Gulsevin, Alican}, year={2023}, month=jan, pages={111-119.e2} }\n",
      "\n",
      " @article{del_Alamo_2022, title={Sampling alternative conformational states of transporters and receptors with AlphaFold2}, volume={11}, ISSN={2050-084X}, url={http://dx.doi.org/10.7554/eLife.75751}, DOI={10.7554/elife.75751}, journal={eLife}, publisher={eLife Sciences Publications, Ltd}, author={del Alamo, Diego and Sala, Davide and Mchaourab, Hassane S and Meiler, Jens}, year={2022}, month=mar }\n",
      "\n",
      " @article{golkov_becker_plop_Äuturilo_davoudi_mendenhall_moretti_meiler_cremers_2020, title={Deep Learning for Virtual Screening: Five Reasons to Use ROC Cost Functions}, url={https://arxiv.org/abs/2007.07029}, DOI={10.48550/ARXIV.2007.07029}, abstractNote={Computer-aided drug discovery is an essential component of modern drug development. Therein, deep learning has become an important tool for rapid screening of billions of molecules in silico for potential hits containing desired chemical features. Despite its importance, substantial challenges persist in training these models, such as severe class imbalance, high decision thresholds, and lack of ground truth labels in some datasets. In this work we argue in favor of directly optimizing the receiver operating characteristic (ROC) in such cases, due to its robustness to class imbalance, its ability to compromise over different decision thresholds, certain freedom to influence the relative weights in this compromise, fidelity to typical benchmarking measures, and equivalence to positive/unlabeled learning. We also propose new training schemes (coherent mini-batch arrangement, and usage of out-of-batch samples) for cost functions based on the ROC, as well as a cost function based on the logAUC metric that facilitates early enrichment (i.e. improves performance at high decision thresholds, as often desired when synthesizing predicted hit compounds). We demonstrate that these approaches outperform standard deep learning approaches on a series of PubChem high-throughput screening datasets that represent realistic and diverse drug discovery campaigns on major drug target families.}, publisher={arXiv}, author={Golkov, Vladimir and Becker, Alexander and Plop, Daniel T. and Äuturilo, Daniel and Davoudi, Neda and Mendenhall, Jeffrey and Moretti, Rocco and Meiler, Jens and Cremers, Daniel}, year={2020} }\n",
      " @article{Sala_2023, title={Modeling conformational states of proteins with AlphaFold}, volume={81}, ISSN={0959-440X}, url={http://dx.doi.org/10.1016/j.sbi.2023.102645}, DOI={10.1016/j.sbi.2023.102645}, journal={Current Opinion in Structural Biology}, publisher={Elsevier BV}, author={Sala, D. and Engelberger, F. and Mchaourab, H.S. and Meiler, J.}, year={2023}, month=aug, pages={102645} }\n",
      "\n",
      " @article{McDonald_2024, title={Benchmarking AlphaMissense pathogenicity predictions against cystic fibrosis variants}, volume={19}, ISSN={1932-6203}, url={http://dx.doi.org/10.1371/journal.pone.0297560}, DOI={10.1371/journal.pone.0297560}, number={1}, journal={PLOS ONE}, publisher={Public Library of Science (PLoS)}, author={McDonald, Eli Fritz and Oliver, Kathryn E. and Schlebach, Jonathan P. and Meiler, Jens and Plate, Lars}, editor={Brodsky, Jeffrey L}, year={2024}, month=jan, pages={e0297560} }\n",
      "\n",
      " @article{Brown_2021, title={General Purpose Structure-Based Drug Discovery Neural Network Score Functions with Human-Interpretable Pharmacophore Maps}, volume={61}, ISSN={1549-960X}, url={http://dx.doi.org/10.1021/acs.jcim.0c01001}, DOI={10.1021/acs.jcim.0c01001}, number={2}, journal={Journal of Chemical Information and Modeling}, publisher={American Chemical Society (ACS)}, author={Brown, Benjamin P. and Mendenhall, Jeffrey and Geanes, Alexander R. and Meiler, Jens}, year={2021}, month=jan, pages={603Ã¢ÂÂ620} }\n",
      "\n",
      " @article{Butkiewicz_2009, title={Application of machine learning approaches on quantitative structure activity relationships}, url={http://dx.doi.org/10.1109/CIBCB.2009.4925736}, DOI={10.1109/cibcb.2009.4925736}, journal={2009 IEEE Symposium on Computational Intelligence in Bioinformatics and Computational Biology}, publisher={IEEE}, author={Butkiewicz, Mariusz and Mueller, Ralf and Selic, Danilo and Dawson, Eric and Meiler, Jens}, year={2009}, month=mar, pages={255Ã¢ÂÂ262} }\n",
      "\n",
      " @article{Brown_2024, title={Approximating Projections of Conformational Boltzmann Distributions with AlphaFold2 Predictions: Opportunities and Limitations}, volume={20}, ISSN={1549-9626}, url={http://dx.doi.org/10.1021/acs.jctc.3c01081}, DOI={10.1021/acs.jctc.3c01081}, number={3}, journal={Journal of Chemical Theory and Computation}, publisher={American Chemical Society (ACS)}, author={Brown, Benjamin P. and Stein, Richard A. and Meiler, Jens and Mchaourab, Hassane S.}, year={2024}, month=jan, pages={1434Ã¢ÂÂ1447} }\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for doi in ai_dois:\n",
    "    print(session.doi2bibtex(doi))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "publication_list",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
